


<!DOCTYPE html>
<html lang="en" class="">
  <head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# object: http://ogp.me/ns/object# article: http://ogp.me/ns/article# profile: http://ogp.me/ns/profile#">
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="Content-Language" content="en">
    
    
    <title>s4vdpca/README.md at master · mwsill/s4vdpca · GitHub</title>
    <link rel="search" type="application/opensearchdescription+xml" href="/opensearch.xml" title="GitHub">
    <link rel="fluid-icon" href="https://github.com/fluidicon.png" title="GitHub">
    <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-114.png">
    <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114.png">
    <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-144.png">
    <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144.png">
    <meta property="fb:app_id" content="1401488693436528">

      <meta content="@github" name="twitter:site" /><meta content="summary" name="twitter:card" /><meta content="mwsill/s4vdpca" name="twitter:title" /><meta content="Contribute to s4vdpca development by creating an account on GitHub." name="twitter:description" /><meta content="https://avatars0.githubusercontent.com/u/8298453?v=3&amp;s=400" name="twitter:image:src" />
      <meta content="GitHub" property="og:site_name" /><meta content="object" property="og:type" /><meta content="https://avatars0.githubusercontent.com/u/8298453?v=3&amp;s=400" property="og:image" /><meta content="mwsill/s4vdpca" property="og:title" /><meta content="https://github.com/mwsill/s4vdpca" property="og:url" /><meta content="Contribute to s4vdpca development by creating an account on GitHub." property="og:description" />
      <meta name="browser-stats-url" content="/_stats">
    <link rel="assets" href="https://assets-cdn.github.com/">
    
    <meta name="pjax-timeout" content="1000">
    

    <meta name="msapplication-TileImage" content="/windows-tile.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="selected-link" value="repo_source" data-pjax-transient>
      <meta name="google-analytics" content="UA-3769691-2">

    <meta content="collector.githubapp.com" name="octolytics-host" /><meta content="collector-cdn.github.com" name="octolytics-script-host" /><meta content="github" name="octolytics-app-id" /><meta content="C1BEC101:666D:6CFB4DD:55104F56" name="octolytics-dimension-request_id" />
    
    <meta content="Rails, view, blob#show" name="analytics-event" />

    
    <link rel="icon" type="image/x-icon" href="https://assets-cdn.github.com/favicon.ico">


    <meta content="authenticity_token" name="csrf-param" />
<meta content="sr/51FDAmK3A1rcmoykzjOTlzGJL7/nzKuZIv9pa133EqL82IVCTQqmh+Am4sPgDDEX2io1GImXtAGGmF3se2w==" name="csrf-token" />

    <link href="https://assets-cdn.github.com/assets/github-099e0ecc2851c8aca89ef6dafa191df3b0f2a2c8ad34e134c5473ca1ba0a59b2.css" media="all" rel="stylesheet" />
    <link href="https://assets-cdn.github.com/assets/github2-40d9bf14363443ccf64a2b71208f59e8739d6288d962feba121227e0608772f3.css" media="all" rel="stylesheet" />
    
    


    <meta http-equiv="x-pjax-version" content="b6b9548efafbc4235d9cb55e3ba0c038">

      
  <meta name="description" content="Contribute to s4vdpca development by creating an account on GitHub.">
  <meta name="go-import" content="github.com/mwsill/s4vdpca git https://github.com/mwsill/s4vdpca.git">

  <meta content="8298453" name="octolytics-dimension-user_id" /><meta content="mwsill" name="octolytics-dimension-user_login" /><meta content="22381665" name="octolytics-dimension-repository_id" /><meta content="mwsill/s4vdpca" name="octolytics-dimension-repository_nwo" /><meta content="true" name="octolytics-dimension-repository_public" /><meta content="false" name="octolytics-dimension-repository_is_fork" /><meta content="22381665" name="octolytics-dimension-repository_network_root_id" /><meta content="mwsill/s4vdpca" name="octolytics-dimension-repository_network_root_nwo" />
  <link href="https://github.com/mwsill/s4vdpca/commits/master.atom" rel="alternate" title="Recent Commits to s4vdpca:master" type="application/atom+xml">

  </head>


  <body class="logged_out  env-production  vis-public page-blob">
    <a href="#start-of-content" tabindex="1" class="accessibility-aid js-skip-to-content">Skip to content</a>
    <div class="wrapper">
      
      
      


        
        <div class="header header-logged-out" role="banner">
  <div class="container clearfix">

    <a class="header-logo-wordmark" href="https://github.com/" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark">
      <span class="mega-octicon octicon-logo-github"></span>
    </a>

    <div class="header-actions" role="navigation">
        <a class="btn btn-primary" href="/join" data-ga-click="(Logged out) Header, clicked Sign up, text:sign-up">Sign up</a>
      <a class="btn" href="/login?return_to=%2Fmwsill%2Fs4vdpca%2Fblob%2Fmaster%2FREADME.md" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in">Sign in</a>
    </div>

    <div class="site-search repo-scope js-site-search" role="search">
      <form accept-charset="UTF-8" action="/mwsill/s4vdpca/search" class="js-site-search-form" data-global-search-url="/search" data-repo-search-url="/mwsill/s4vdpca/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
  <input type="text"
    class="js-site-search-field is-clearable"
    data-hotkey="s"
    name="q"
    placeholder="Search"
    data-global-scope-placeholder="Search GitHub"
    data-repo-scope-placeholder="Search"
    tabindex="1"
    autocapitalize="off">
  <div class="scope-badge">This repository</div>
</form>
    </div>

      <ul class="header-nav left" role="navigation">
          <li class="header-nav-item">
            <a class="header-nav-link" href="/explore" data-ga-click="(Logged out) Header, go to explore, text:explore">Explore</a>
          </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="/features" data-ga-click="(Logged out) Header, go to features, text:features">Features</a>
          </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="https://enterprise.github.com/" data-ga-click="(Logged out) Header, go to enterprise, text:enterprise">Enterprise</a>
          </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="/blog" data-ga-click="(Logged out) Header, go to blog, text:blog">Blog</a>
          </li>
      </ul>

  </div>
</div>



      <div id="start-of-content" class="accessibility-aid"></div>
          <div class="site" itemscope itemtype="http://schema.org/WebPage">
    <div id="js-flash-container">
      
    </div>
    <div class="pagehead repohead instapaper_ignore readability-menu">
      <div class="container">
        
<ul class="pagehead-actions">

  <li>
      <a href="/login?return_to=%2Fmwsill%2Fs4vdpca"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to watch a repository" rel="nofollow">
    <span class="octicon octicon-eye"></span>
    Watch
  </a>
  <a class="social-count" href="/mwsill/s4vdpca/watchers">
    1
  </a>

  </li>

  <li>
      <a href="/login?return_to=%2Fmwsill%2Fs4vdpca"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to star a repository" rel="nofollow">
    <span class="octicon octicon-star"></span>
    Star
  </a>

    <a class="social-count js-social-count" href="/mwsill/s4vdpca/stargazers">
      1
    </a>

  </li>

    <li>
      <a href="/login?return_to=%2Fmwsill%2Fs4vdpca"
        class="btn btn-sm btn-with-count tooltipped tooltipped-n"
        aria-label="You must be signed in to fork a repository" rel="nofollow">
        <span class="octicon octicon-repo-forked"></span>
        Fork
      </a>
      <a href="/mwsill/s4vdpca/network" class="social-count">
        0
      </a>
    </li>
</ul>

        <h1 itemscope itemtype="http://data-vocabulary.org/Breadcrumb" class="entry-title public">
          <span class="mega-octicon octicon-repo"></span>
          <span class="author"><a href="/mwsill" class="url fn" itemprop="url" rel="author"><span itemprop="title">mwsill</span></a></span><!--
       --><span class="path-divider">/</span><!--
       --><strong><a href="/mwsill/s4vdpca" class="js-current-repository" data-pjax="#js-repo-pjax-container">s4vdpca</a></strong>

          <span class="page-context-loader">
            <img alt="" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
          </span>

        </h1>
      </div><!-- /.container -->
    </div><!-- /.repohead -->

    <div class="container">
      <div class="repository-with-sidebar repo-container new-discussion-timeline  ">
        <div class="repository-sidebar clearfix">
            
<nav class="sunken-menu repo-nav js-repo-nav js-sidenav-container-pjax js-octicon-loaders"
     role="navigation"
     data-pjax="#js-repo-pjax-container"
     data-issue-count-url="/mwsill/s4vdpca/issues/counts">
  <ul class="sunken-menu-group">
    <li class="tooltipped tooltipped-w" aria-label="Code">
      <a href="/mwsill/s4vdpca" aria-label="Code" class="selected js-selected-navigation-item sunken-menu-item" data-hotkey="g c" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches /mwsill/s4vdpca">
        <span class="octicon octicon-code"></span> <span class="full-word">Code</span>
        <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>

      <li class="tooltipped tooltipped-w" aria-label="Issues">
        <a href="/mwsill/s4vdpca/issues" aria-label="Issues" class="js-selected-navigation-item sunken-menu-item" data-hotkey="g i" data-selected-links="repo_issues repo_labels repo_milestones /mwsill/s4vdpca/issues">
          <span class="octicon octicon-issue-opened"></span> <span class="full-word">Issues</span>
          <span class="js-issue-replace-counter"></span>
          <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>      </li>

    <li class="tooltipped tooltipped-w" aria-label="Pull requests">
      <a href="/mwsill/s4vdpca/pulls" aria-label="Pull requests" class="js-selected-navigation-item sunken-menu-item" data-hotkey="g p" data-selected-links="repo_pulls /mwsill/s4vdpca/pulls">
          <span class="octicon octicon-git-pull-request"></span> <span class="full-word">Pull requests</span>
          <span class="js-pull-replace-counter"></span>
          <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>


  </ul>
  <div class="sunken-menu-separator"></div>
  <ul class="sunken-menu-group">

    <li class="tooltipped tooltipped-w" aria-label="Pulse">
      <a href="/mwsill/s4vdpca/pulse" aria-label="Pulse" class="js-selected-navigation-item sunken-menu-item" data-selected-links="pulse /mwsill/s4vdpca/pulse">
        <span class="octicon octicon-pulse"></span> <span class="full-word">Pulse</span>
        <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>

    <li class="tooltipped tooltipped-w" aria-label="Graphs">
      <a href="/mwsill/s4vdpca/graphs" aria-label="Graphs" class="js-selected-navigation-item sunken-menu-item" data-selected-links="repo_graphs repo_contributors /mwsill/s4vdpca/graphs">
        <span class="octicon octicon-graph"></span> <span class="full-word">Graphs</span>
        <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>
  </ul>


</nav>

              <div class="only-with-full-nav">
                  
<div class="clone-url open"
  data-protocol-type="http"
  data-url="/users/set_protocol?protocol_selector=http&amp;protocol_type=clone">
  <h3><span class="text-emphasized">HTTPS</span> clone URL</h3>
  <div class="input-group js-zeroclipboard-container">
    <input type="text" class="input-mini input-monospace js-url-field js-zeroclipboard-target"
           value="https://github.com/mwsill/s4vdpca.git" readonly="readonly">
    <span class="input-group-button">
      <button aria-label="Copy to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button" data-copied-hint="Copied!" type="button"><span class="octicon octicon-clippy"></span></button>
    </span>
  </div>
</div>

  
<div class="clone-url "
  data-protocol-type="subversion"
  data-url="/users/set_protocol?protocol_selector=subversion&amp;protocol_type=clone">
  <h3><span class="text-emphasized">Subversion</span> checkout URL</h3>
  <div class="input-group js-zeroclipboard-container">
    <input type="text" class="input-mini input-monospace js-url-field js-zeroclipboard-target"
           value="https://github.com/mwsill/s4vdpca" readonly="readonly">
    <span class="input-group-button">
      <button aria-label="Copy to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button" data-copied-hint="Copied!" type="button"><span class="octicon octicon-clippy"></span></button>
    </span>
  </div>
</div>



<p class="clone-options">You can clone with
  <a href="#" class="js-clone-selector" data-protocol="http">HTTPS</a> or <a href="#" class="js-clone-selector" data-protocol="subversion">Subversion</a>.
  <a href="https://help.github.com/articles/which-remote-url-should-i-use" class="help tooltipped tooltipped-n" aria-label="Get help on which URL is right for you.">
    <span class="octicon octicon-question"></span>
  </a>
</p>



                <a href="/mwsill/s4vdpca/archive/master.zip"
                   class="btn btn-sm sidebar-button"
                   aria-label="Download the contents of mwsill/s4vdpca as a zip file"
                   title="Download the contents of mwsill/s4vdpca as a zip file"
                   rel="nofollow">
                  <span class="octicon octicon-cloud-download"></span>
                  Download ZIP
                </a>
              </div>
        </div><!-- /.repository-sidebar -->

        <div id="js-repo-pjax-container" class="repository-content context-loader-container" data-pjax-container>
          

<a href="/mwsill/s4vdpca/blob/d7e48f1891597faa5dddfa51759f9f1e0ba8dea0/README.md" class="hidden js-permalink-shortcut" data-hotkey="y">Permalink</a>

<!-- blob contrib key: blob_contributors:v21:51f512a1f6cbf10ff2106c3e4d5c3c26 -->

<div class="file-navigation js-zeroclipboard-container">
  
<div class="select-menu js-menu-container js-select-menu left">
  <span class="btn btn-sm select-menu-button js-menu-target css-truncate" data-hotkey="w"
    data-master-branch="master"
    data-ref="master"
    title="master"
    role="button" aria-label="Switch branches or tags" tabindex="0" aria-haspopup="true">
    <span class="octicon octicon-git-branch"></span>
    <i>branch:</i>
    <span class="js-select-button css-truncate-target">master</span>
  </span>

  <div class="select-menu-modal-holder js-menu-content js-navigation-container" data-pjax aria-hidden="true">

    <div class="select-menu-modal">
      <div class="select-menu-header">
        <span class="select-menu-title">Switch branches/tags</span>
        <span class="octicon octicon-x js-menu-close" role="button" aria-label="Close"></span>
      </div>

      <div class="select-menu-filters">
        <div class="select-menu-text-filter">
          <input type="text" aria-label="Filter branches/tags" id="context-commitish-filter-field" class="js-filterable-field js-navigation-enable" placeholder="Filter branches/tags">
        </div>
        <div class="select-menu-tabs">
          <ul>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="branches" data-filter-placeholder="Filter branches/tags" class="js-select-menu-tab">Branches</a>
            </li>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="tags" data-filter-placeholder="Find a tag…" class="js-select-menu-tab">Tags</a>
            </li>
          </ul>
        </div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="branches">

        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


            <a class="select-menu-item js-navigation-item js-navigation-open selected"
               href="/mwsill/s4vdpca/blob/master/README.md"
               data-name="master"
               data-skip-pjax="true"
               rel="nofollow">
              <span class="select-menu-item-icon octicon octicon-check"></span>
              <span class="select-menu-item-text css-truncate-target" title="master">
                master
              </span>
            </a>
        </div>

          <div class="select-menu-no-results">Nothing to show</div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="tags">
        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


        </div>

        <div class="select-menu-no-results">Nothing to show</div>
      </div>

    </div>
  </div>
</div>

  <div class="btn-group right">
    <a href="/mwsill/s4vdpca/find/master"
          class="js-show-file-finder btn btn-sm empty-icon tooltipped tooltipped-s"
          data-pjax
          data-hotkey="t"
          aria-label="Quickly jump between files">
      <span class="octicon octicon-list-unordered"></span>
    </a>
    <button aria-label="Copy file path to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button" data-copied-hint="Copied!" type="button"><span class="octicon octicon-clippy"></span></button>
  </div>

  <div class="breadcrumb js-zeroclipboard-target">
    <span class='repo-root js-repo-root'><span itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a href="/mwsill/s4vdpca" class="" data-branch="master" data-direction="back" data-pjax="true" itemscope="url"><span itemprop="title">s4vdpca</span></a></span></span><span class="separator">/</span><strong class="final-path">README.md</strong>
  </div>
</div>

<include-fragment class="commit commit-loader file-history-tease" src="/mwsill/s4vdpca/contributors/master/README.md">
  <div class="file-history-tease-header">
    Fetching contributors&hellip;
  </div>

  <div class="participation">
    <p class="loader-loading"><img alt="" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-EAF2F5-0bdc57d34b85c4a4de9d0d1db10cd70e8a95f33ff4f46c5a8c48b4bf4e5a9abe.gif" width="16" /></p>
    <p class="loader-error">Cannot retrieve contributors at this time</p>
  </div>
</include-fragment>
<div class="file">
  <div class="file-header">
    <div class="file-actions">

      <div class="btn-group">
        <a href="/mwsill/s4vdpca/raw/master/README.md" class="btn btn-sm " id="raw-url">Raw</a>
          <a href="/mwsill/s4vdpca/blame/master/README.md" class="btn btn-sm js-update-url-with-hash">Blame</a>
        <a href="/mwsill/s4vdpca/commits/master/README.md" class="btn btn-sm " rel="nofollow">History</a>
      </div>


          <button type="button" class="octicon-btn disabled tooltipped tooltipped-n" aria-label="You must be signed in to make or propose changes">
            <span class="octicon octicon-pencil"></span>
          </button>

        <button type="button" class="octicon-btn octicon-btn-danger disabled tooltipped tooltipped-n" aria-label="You must be signed in to make or propose changes">
          <span class="octicon octicon-trashcan"></span>
        </button>
    </div>

    <div class="file-info">
        316 lines (272 sloc)
        <span class="file-info-divider"></span>
      10.7 kb
    </div>
  </div>
    <div id="readme" class="blob instapaper_body">
    <article class="markdown-body entry-content" itemprop="mainContentOfPage"><h1>
<a id="user-content-s4vdpca" class="anchor" href="#s4vdpca" aria-hidden="true"><span class="octicon octicon-link"></span></a>s4vdpca</h1>

<p>This R package implements methods for sparse principal component analysis as descriped in the manuscript
"Applying Stability Selection to Consistently Estimate Sparse Principal Components in High-Dimensional Molecular Data" submitted to Oxford Bioinformatics.</p>

<div class="highlight highlight-r"><pre><span class="pl-c"># install package using devtools</span>
<span class="pl-c"># install.packages('devtools')</span>
library(<span class="pl-smi">devtools</span>)                  
install_github(<span class="pl-s"><span class="pl-pds">'</span>mwsill/s4vdpca<span class="pl-pds">'</span></span>)
library(<span class="pl-smi">s4vdpca</span>)

<span class="pl-c"># generate a simulated data set using the single-covariance spike model </span>
<span class="pl-smi">p</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">5000</span>    <span class="pl-c"># number of variables</span>
<span class="pl-smi">n</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">100</span>      <span class="pl-c"># number of observations</span>
<span class="pl-smi">alpha</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">0.8</span>  <span class="pl-c"># spike index </span>
<span class="pl-smi">beta</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">0.8</span>   <span class="pl-c"># sparsity index </span>

<span class="pl-c"># generate a population variance covariance matrix</span>
<span class="pl-smi">Sigma</span> <span class="pl-k">&lt;-</span> generate_covar(<span class="pl-smi">alpha</span>,<span class="pl-smi">beta</span>,<span class="pl-smi">p</span>)

<span class="pl-c"># extract first eigenvector</span>
<span class="pl-smi">z1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">Sigma</span>[[<span class="pl-c1">2</span>]]

<span class="pl-c"># extract variance covariance matrix</span>
<span class="pl-smi">Sigma</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">Sigma</span>[[<span class="pl-c1">1</span>]]

<span class="pl-c"># sample from multivariate normal distribution using Cholesky decomposition</span>
<span class="pl-c"># see ?rmvn in package broman for details</span>
<span class="pl-smi">D</span> <span class="pl-k">&lt;-</span> chol(<span class="pl-smi">Sigma</span>)
set.seed(<span class="pl-c1">24022015</span>)
<span class="pl-smi">x</span> <span class="pl-k">&lt;-</span> <span class="pl-k">matrix</span>(rnorm(<span class="pl-smi">n</span> <span class="pl-k">*</span> <span class="pl-smi">p</span>), <span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-smi">p</span>) <span class="pl-k">%*%</span> <span class="pl-smi">D</span> <span class="pl-k">+</span> rep(rep(<span class="pl-c1">0</span>,<span class="pl-smi">p</span>), rep(<span class="pl-smi">n</span>, <span class="pl-smi">p</span>))

<span class="pl-c">#show documentation</span>
?<span class="pl-smi">s4vdpca</span>
?<span class="pl-smi">rspca</span>

<span class="pl-c"># apply S4VDPCA and RSPCA with different penalization functions, all with GIC5 </span>
<span class="pl-c"># parallelization is not yet available on Windows machines</span>
<span class="pl-smi">res1</span> <span class="pl-k">&lt;-</span> s4vdpca(<span class="pl-smi">x</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>gic5<span class="pl-pds">'</span></span>,<span class="pl-v">weakness</span><span class="pl-k">=</span><span class="pl-c1">0.2</span>,<span class="pl-v">B</span><span class="pl-k">=</span><span class="pl-c1">500</span>)
<span class="pl-smi">res2</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">x</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>gic5<span class="pl-pds">'</span></span>) <span class="pl-c">#lasso</span>
<span class="pl-smi">res3</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">x</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>gic5<span class="pl-pds">'</span></span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>scad<span class="pl-pds">'</span></span>) <span class="pl-c">#scad </span>
<span class="pl-smi">res4</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">x</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>gic5<span class="pl-pds">'</span></span>, <span class="pl-v">gamv</span><span class="pl-k">=</span><span class="pl-c1">1</span>) <span class="pl-c"># adaptive lasso</span>

<span class="pl-c"># plot the information criterion</span>
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">2</span>))
plot(<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">ic</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>number of selected features<span class="pl-pds">'</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>GIC 5<span class="pl-pds">'</span></span>
,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>S4VDPCA<span class="pl-pds">'</span></span>)
abline(<span class="pl-v">v</span><span class="pl-k">=</span><span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">minic</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)
text(<span class="pl-v">y</span><span class="pl-k">=</span>max(<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">ic</span>,<span class="pl-v">na.rm</span><span class="pl-k">=</span><span class="pl-c1">T</span>)<span class="pl-k">-</span><span class="pl-c1">1000</span>,<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">minic</span><span class="pl-k">+</span><span class="pl-c1">100</span>,<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)

plot(<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">ic</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>number of selected features<span class="pl-pds">'</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>GIC 5<span class="pl-pds">'</span></span>
,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>RSPCA lasso<span class="pl-pds">'</span></span>)
abline(<span class="pl-v">v</span><span class="pl-k">=</span><span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">minic</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)
text(<span class="pl-v">y</span><span class="pl-k">=</span>max(<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">ic</span>,<span class="pl-v">na.rm</span><span class="pl-k">=</span><span class="pl-c1">T</span>)<span class="pl-k">-</span><span class="pl-c1">1000</span>,<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">minic</span><span class="pl-k">+</span><span class="pl-c1">100</span>,<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)

plot(<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">ic</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>number of selected features<span class="pl-pds">'</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>GIC 5<span class="pl-pds">'</span></span>
,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>RSPCA scad<span class="pl-pds">'</span></span>)
abline(<span class="pl-v">v</span><span class="pl-k">=</span><span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">minic</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)
text(<span class="pl-v">y</span><span class="pl-k">=</span>max(<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">ic</span>,<span class="pl-v">na.rm</span><span class="pl-k">=</span><span class="pl-c1">T</span>)<span class="pl-k">-</span><span class="pl-c1">1000</span>,<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">minic</span><span class="pl-k">+</span><span class="pl-c1">100</span>,<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)

plot(<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">ic</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>number of selected features<span class="pl-pds">'</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>GIC 5<span class="pl-pds">'</span></span>
,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>RSPCA adaptive lasso<span class="pl-pds">'</span></span>)
abline(<span class="pl-v">v</span><span class="pl-k">=</span><span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">minic</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)
text(<span class="pl-v">y</span><span class="pl-k">=</span>max(<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">ic</span>,<span class="pl-v">na.rm</span><span class="pl-k">=</span><span class="pl-c1">T</span>)<span class="pl-k">-</span><span class="pl-c1">1000</span>,<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">minic</span><span class="pl-k">+</span><span class="pl-c1">100</span>,<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>red<span class="pl-pds">'</span></span>)

<span class="pl-c"># calculate angle between estimated sparse loadings vector and simulated eigenvector</span>
angle(<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">v</span>,<span class="pl-smi">z1</span>)
angle(<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">v</span>,<span class="pl-smi">z1</span>)
angle(<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">v</span>,<span class="pl-smi">z1</span>)
angle(<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">v</span>,<span class="pl-smi">z1</span>)

<span class="pl-c"># calculate number of falsely selected features</span>
type1(<span class="pl-smi">z1</span>,<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
type1(<span class="pl-smi">z1</span>,<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
type1(<span class="pl-smi">z1</span>,<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
type1(<span class="pl-smi">z1</span>,<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">v</span>)

<span class="pl-c"># calculate number of type2 errors</span>
type2(<span class="pl-smi">z1</span>,<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
type2(<span class="pl-smi">z1</span>,<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
type2(<span class="pl-smi">z1</span>,<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
type2(<span class="pl-smi">z1</span>,<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">v</span>)

<span class="pl-c"># calculate empirical spike index</span>
emp_spike(<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">sdev</span>,<span class="pl-smi">p</span>)
emp_spike(<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">sdev</span>,<span class="pl-smi">p</span>)
emp_spike(<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">sdev</span>,<span class="pl-smi">p</span>)
emp_spike(<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">sdev</span>,<span class="pl-smi">p</span>)
<span class="pl-smi">alpha</span>

<span class="pl-c"># calculate empirical sparsity index</span>
emp_sparsity(<span class="pl-smi">res1</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-smi">p</span>)
emp_sparsity(<span class="pl-smi">res2</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-smi">p</span>)
emp_sparsity(<span class="pl-smi">res3</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-smi">p</span>)
emp_sparsity(<span class="pl-smi">res4</span><span class="pl-k">$</span><span class="pl-smi">minic</span>,<span class="pl-smi">p</span>)
<span class="pl-smi">beta</span>

<span class="pl-c"># apply regular PCA and calculate angle between loadings vector</span>
<span class="pl-c"># and simulated eigenvector</span>
<span class="pl-smi">pca</span> <span class="pl-k">&lt;-</span> prcomp(<span class="pl-smi">x</span>)
angle(<span class="pl-smi">pca</span><span class="pl-k">$</span><span class="pl-smi">rotation</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">z1</span>)

<span class="pl-c"># ssvdpca is the original rspca function by Lee et al. 2010</span>
<span class="pl-smi">X</span>  <span class="pl-k">&lt;-</span> scale(<span class="pl-smi">x</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
system.time(
<span class="pl-smi">res6</span> <span class="pl-k">&lt;-</span> ssvdpca(<span class="pl-smi">X</span>) <span class="pl-c">#lasso</span>
)
<span class="pl-c"># optimized code; search for minimal bic</span>
system.time(
<span class="pl-smi">res7</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">X</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">1</span>,<span class="pl-v">steps</span><span class="pl-k">=</span><span class="pl-c1">100</span>, <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>bic<span class="pl-pds">'</span></span>) <span class="pl-c">#lasso</span>
)
<span class="pl-c"># optimized code; parallelized search for minimal bic, only on Unix machines</span>
system.time(
<span class="pl-smi">res8</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">X</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">steps</span><span class="pl-k">=</span><span class="pl-c1">100</span>, <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>bic<span class="pl-pds">'</span></span>) <span class="pl-c">#lasso</span>
)
<span class="pl-c"># estimated loadings are the same </span>
all(<span class="pl-smi">res6</span><span class="pl-k">$</span><span class="pl-smi">v</span><span class="pl-k">==</span><span class="pl-smi">res7</span><span class="pl-k">$</span><span class="pl-smi">v</span>)
all(<span class="pl-smi">res6</span><span class="pl-k">$</span><span class="pl-smi">v</span><span class="pl-k">==</span><span class="pl-smi">res8</span><span class="pl-k">$</span><span class="pl-smi">v</span>)


<span class="pl-c">###### real data application example</span>
<span class="pl-c"># to perform the analysis the follwing Bioconductor packages are needed</span>
<span class="pl-c">#source("http://bioconductor.org/biocLite.R")</span>
<span class="pl-c">#biocLite(c("Biobase","HTSanalyzeR","org.Hs.eg.db","KEGG.db"))</span>

<span class="pl-c"># load packages</span>
library(<span class="pl-smi">Biobase</span>)
library(<span class="pl-smi">KEGG.db</span>)
library(<span class="pl-smi">org.Hs.eg.db</span>)
library(<span class="pl-smi">HTSanalyzeR</span>)

<span class="pl-c"># load medulloblastoma example data set </span>
data(<span class="pl-smi">medullo</span>)

<span class="pl-c"># data is stored as ExpressionSet</span>
<span class="pl-smi">medullo</span>

<span class="pl-c"># extract gene expression matrix</span>
<span class="pl-smi">X</span> <span class="pl-k">&lt;-</span> scale(t(exprs(<span class="pl-smi">medullo</span>)),<span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
dim(<span class="pl-smi">X</span>)

<span class="pl-c"># estimate first sparse PC using S4VDPCA</span>
set.seed(<span class="pl-c1">10102014</span>,<span class="pl-s"><span class="pl-pds">"</span>L'Ecuyer-CMRG<span class="pl-pds">"</span></span>)
<span class="pl-smi">res.s4vdpca</span> <span class="pl-k">&lt;-</span> s4vdpca(<span class="pl-smi">X</span>,<span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>,<span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">steps</span><span class="pl-k">=</span><span class="pl-c1">1000</span>,<span class="pl-v">B</span><span class="pl-k">=</span><span class="pl-c1">500</span>) 
<span class="pl-smi">res.rspca</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">X</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">steps</span><span class="pl-k">=</span><span class="pl-c1">1000</span>,<span class="pl-v">niter</span><span class="pl-k">=</span><span class="pl-c1">200</span>,<span class="pl-v">merr</span><span class="pl-k">=</span><span class="pl-c1">10</span><span class="pl-k">^</span>(<span class="pl-k">-</span><span class="pl-c1">4</span>), <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>gic5<span class="pl-pds">'</span></span>) 
<span class="pl-c"># subtract first sparse rank1 layer</span>
<span class="pl-smi">X.1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">X</span> <span class="pl-k">-</span> (<span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">d</span> <span class="pl-k">*</span> <span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">u</span> <span class="pl-k">%*%</span> t(<span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">v</span>)) 
<span class="pl-smi">X.2</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">X</span> <span class="pl-k">-</span> (<span class="pl-smi">res.rspca</span><span class="pl-k">$</span><span class="pl-smi">d</span> <span class="pl-k">*</span> <span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">u</span> <span class="pl-k">%*%</span> t(<span class="pl-smi">res.rspca</span><span class="pl-k">$</span><span class="pl-smi">v</span>) )
<span class="pl-c"># obtain second sparse PC</span>
<span class="pl-smi">res.s4vdpca2</span> <span class="pl-k">&lt;-</span> s4vdpca(<span class="pl-smi">X.1</span>,<span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>,<span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">steps</span><span class="pl-k">=</span><span class="pl-c1">1000</span>,<span class="pl-v">B</span><span class="pl-k">=</span><span class="pl-c1">500</span>)
<span class="pl-smi">res.rspca2</span> <span class="pl-k">&lt;-</span> rspca(<span class="pl-smi">X.2</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">steps</span><span class="pl-k">=</span><span class="pl-c1">1000</span>,<span class="pl-v">niter</span><span class="pl-k">=</span><span class="pl-c1">200</span>,<span class="pl-v">merr</span><span class="pl-k">=</span><span class="pl-c1">10</span><span class="pl-k">^</span>(<span class="pl-k">-</span><span class="pl-c1">4</span>), <span class="pl-v">ic_type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>gic5<span class="pl-pds">'</span></span>) 


<span class="pl-c"># perform gene set over representation analysis for PC1</span>
<span class="pl-smi">t_all</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">v</span>
<span class="pl-c"># get entrez gene ids</span>
names(<span class="pl-smi">t_all</span>) <span class="pl-k">&lt;-</span> featureData(<span class="pl-smi">medullo</span>)<span class="pl-k">$</span><span class="pl-smi">GENE</span>
<span class="pl-c"># define selected genes</span>
<span class="pl-smi">t_hits</span> <span class="pl-k">&lt;-</span> as.character(featureData(<span class="pl-smi">medullo</span>)<span class="pl-k">$</span><span class="pl-smi">GENE</span>[<span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">v</span><span class="pl-k">!=</span><span class="pl-c1">0</span>])
<span class="pl-smi">t_all</span> <span class="pl-k">&lt;-</span> sort(<span class="pl-smi">t_all</span>)
<span class="pl-smi">KEGG_pathways</span> <span class="pl-k">&lt;-</span> KeggGeneSets(<span class="pl-v">species</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Hs<span class="pl-pds">"</span></span>) 
<span class="pl-smi">gscs.kegg</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-smi">KEGG_pathways</span>) 
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> new(<span class="pl-s"><span class="pl-pds">"</span>GSCA<span class="pl-pds">"</span></span>, <span class="pl-v">listOfGeneSetCollections</span><span class="pl-k">=</span><span class="pl-smi">gscs.kegg</span>,
                 <span class="pl-v">geneList</span><span class="pl-k">=</span><span class="pl-smi">t_all</span>, <span class="pl-v">hits</span><span class="pl-k">=</span><span class="pl-smi">t_hits</span>)
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> preprocess(<span class="pl-smi">gsca.kegg</span>, <span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Hs<span class="pl-pds">"</span></span>,
                        <span class="pl-v">initialIDs</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Entrez.gene<span class="pl-pds">"</span></span>,
                        <span class="pl-v">keepMultipleMappings</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,
                        <span class="pl-v">duplicateRemoverMethod</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>max<span class="pl-pds">"</span></span>,
                        <span class="pl-v">orderAbsValue</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> analyze(<span class="pl-smi">gsca.kegg</span>, <span class="pl-v">para</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-v">pValueCutoff</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,
                                          <span class="pl-v">pAdjustMethod</span> <span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>,
                                          <span class="pl-v">nPermutations</span><span class="pl-k">=</span><span class="pl-c1">5</span>,
                                          <span class="pl-v">minGeneSetSize</span><span class="pl-k">=</span><span class="pl-c1">20</span>,
                                          <span class="pl-v">exponent</span><span class="pl-k">=</span><span class="pl-c1">1</span>),
                      <span class="pl-v">verbose</span><span class="pl-k">=</span><span class="pl-c1">T</span>, <span class="pl-v">doGSEA</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> appendGSTerms(<span class="pl-smi">gsca.kegg</span>,
                           <span class="pl-v">keggGSCs</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span>)
length(getTopGeneSets(<span class="pl-smi">gsca.kegg</span>,<span class="pl-s"><span class="pl-pds">"</span>HyperGeo.results<span class="pl-pds">"</span></span>,
                      <span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span>,<span class="pl-v">allSig</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)[[<span class="pl-c1">1</span>]])
<span class="pl-smi">hypPC1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">gsca.kegg</span><span class="pl-k">@</span><span class="pl-smi">result</span><span class="pl-k">$</span><span class="pl-smi">HyperGeo.results</span><span class="pl-k">$</span>`<span class="pl-smi">KEGG</span> <span class="pl-smi">pathways</span>`
<span class="pl-smi">g1</span> <span class="pl-k">&lt;-</span> viewEnrichMap(<span class="pl-smi">gsca.kegg</span>, <span class="pl-v">resultName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>HyperGeo.results<span class="pl-pds">"</span></span>, <span class="pl-v">gscs</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span>,
                    <span class="pl-v">ntop</span><span class="pl-k">=</span><span class="pl-c1">6</span>, <span class="pl-v">allSig</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">gsNameType</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>term<span class="pl-pds">"</span></span>, <span class="pl-v">displayEdgeLabel</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>,
                    <span class="pl-v">layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>layout.kamada.kawai<span class="pl-pds">"</span></span>,<span class="pl-v">plot</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

<span class="pl-c"># perform gene set over representation analysis for PC2</span>
<span class="pl-smi">t_all</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">res.s4vdpca2</span><span class="pl-k">$</span><span class="pl-smi">v</span>
<span class="pl-c"># get entrez gene ids</span>
names(<span class="pl-smi">t_all</span>) <span class="pl-k">&lt;-</span> featureData(<span class="pl-smi">medullo</span>)<span class="pl-k">$</span><span class="pl-smi">GENE</span>
<span class="pl-c"># define selected genes</span>
<span class="pl-smi">t_hits</span> <span class="pl-k">&lt;-</span> as.character(featureData(<span class="pl-smi">medullo</span>)<span class="pl-k">$</span><span class="pl-smi">GENE</span>[<span class="pl-smi">res.s4vdpca2</span><span class="pl-k">$</span><span class="pl-smi">v</span><span class="pl-k">!=</span><span class="pl-c1">0</span>])
<span class="pl-smi">t_all</span> <span class="pl-k">&lt;-</span> sort(<span class="pl-smi">t_all</span>)
<span class="pl-smi">KEGG_pathways</span> <span class="pl-k">&lt;-</span> KeggGeneSets(<span class="pl-v">species</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Hs<span class="pl-pds">"</span></span>) 
<span class="pl-smi">gscs.kegg</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-smi">KEGG_pathways</span>) 
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> new(<span class="pl-s"><span class="pl-pds">"</span>GSCA<span class="pl-pds">"</span></span>, <span class="pl-v">listOfGeneSetCollections</span><span class="pl-k">=</span><span class="pl-smi">gscs.kegg</span>,
                 <span class="pl-v">geneList</span><span class="pl-k">=</span><span class="pl-smi">t_all</span>, <span class="pl-v">hits</span><span class="pl-k">=</span><span class="pl-smi">t_hits</span>)
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> preprocess(<span class="pl-smi">gsca.kegg</span>, <span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Hs<span class="pl-pds">"</span></span>,
                        <span class="pl-v">initialIDs</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Entrez.gene<span class="pl-pds">"</span></span>,
                        <span class="pl-v">keepMultipleMappings</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,
                        <span class="pl-v">duplicateRemoverMethod</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>max<span class="pl-pds">"</span></span>,
                        <span class="pl-v">orderAbsValue</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> analyze(<span class="pl-smi">gsca.kegg</span>, <span class="pl-v">para</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-v">pValueCutoff</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,
                                          <span class="pl-v">pAdjustMethod</span> <span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>,
                                          <span class="pl-v">nPermutations</span><span class="pl-k">=</span><span class="pl-c1">5</span>,
                                          <span class="pl-v">minGeneSetSize</span><span class="pl-k">=</span><span class="pl-c1">20</span>,
                                          <span class="pl-v">exponent</span><span class="pl-k">=</span><span class="pl-c1">1</span>),
                      <span class="pl-v">verbose</span><span class="pl-k">=</span><span class="pl-c1">T</span>, <span class="pl-v">doGSEA</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-smi">gsca.kegg</span> <span class="pl-k">&lt;-</span> appendGSTerms(<span class="pl-smi">gsca.kegg</span>,
                           <span class="pl-v">keggGSCs</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span>)
length(getTopGeneSets(<span class="pl-smi">gsca.kegg</span>,<span class="pl-s"><span class="pl-pds">"</span>HyperGeo.results<span class="pl-pds">"</span></span>,
                      <span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span>,<span class="pl-v">allSig</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)[[<span class="pl-c1">1</span>]])
<span class="pl-smi">hypPC2</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">gsca.kegg</span><span class="pl-k">@</span><span class="pl-smi">result</span><span class="pl-k">$</span><span class="pl-smi">HyperGeo.results</span><span class="pl-k">$</span>`<span class="pl-smi">KEGG</span> <span class="pl-smi">pathways</span>`
<span class="pl-smi">g2</span> <span class="pl-k">&lt;-</span> viewEnrichMap(<span class="pl-smi">gsca.kegg</span>, <span class="pl-v">resultName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>HyperGeo.results<span class="pl-pds">"</span></span>, <span class="pl-v">gscs</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>KEGG pathways<span class="pl-pds">"</span></span>,
                    <span class="pl-v">ntop</span><span class="pl-k">=</span><span class="pl-c1">6</span>, <span class="pl-v">allSig</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">gsNameType</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>term<span class="pl-pds">"</span></span>, <span class="pl-v">displayEdgeLabel</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>,
                    <span class="pl-v">layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>layout.kamada.kawai<span class="pl-pds">"</span></span>,<span class="pl-v">plot</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)


<span class="pl-c">###### Biplot with dominant marker genes and KEGG pathways</span>

<span class="pl-c"># assign colors to molecular subtypes</span>
<span class="pl-smi">cols</span> <span class="pl-k">&lt;-</span> rep(<span class="pl-s"><span class="pl-pds">"</span>gray<span class="pl-pds">"</span></span>,length(<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span>))
<span class="pl-smi">cols</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>group D<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>darkgreen<span class="pl-pds">"</span></span>
<span class="pl-smi">cols</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>group C<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>
<span class="pl-smi">cols</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>WNT<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>blue<span class="pl-pds">'</span></span> 
<span class="pl-smi">cols</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>SHH<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>darkred<span class="pl-pds">"</span></span>
<span class="pl-c"># assign pch type to molecular subtypes</span>
<span class="pl-smi">ptype</span> <span class="pl-k">&lt;-</span> rep(<span class="pl-c1">1</span>,length(<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span>))
<span class="pl-smi">ptype</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>group D<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">17</span>
<span class="pl-smi">ptype</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>group C<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">15</span>
<span class="pl-smi">ptype</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>WNT<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">18</span>
<span class="pl-smi">ptype</span>[<span class="pl-smi">medullo</span><span class="pl-k">$</span><span class="pl-smi">subtype</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>SHH<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">20</span>
<span class="pl-smi">ptype</span><span class="pl-k">&lt;-</span> as.numeric(<span class="pl-smi">ptype</span>)

<span class="pl-smi">agi2symbol</span> <span class="pl-k">&lt;-</span> featureData(<span class="pl-smi">medullo</span>)<span class="pl-k">$</span><span class="pl-smi">GENE_SYMBOL</span>

<span class="pl-smi">alpha</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">0.37</span>
<span class="pl-smi">ptx</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">u</span><span class="pl-k">*</span>(<span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">d</span><span class="pl-k">^</span><span class="pl-smi">alpha</span>)
<span class="pl-smi">pty</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">res.s4vdpca2</span><span class="pl-k">$</span><span class="pl-smi">u</span><span class="pl-k">*</span>(<span class="pl-smi">res.s4vdpca2</span><span class="pl-k">$</span><span class="pl-smi">d</span><span class="pl-k">^</span><span class="pl-smi">alpha</span>)
<span class="pl-smi">arrowsx</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">v</span><span class="pl-k">*</span>(<span class="pl-smi">res.s4vdpca</span><span class="pl-k">$</span><span class="pl-smi">d</span><span class="pl-k">^</span>(<span class="pl-c1">1</span><span class="pl-k">-</span><span class="pl-smi">alpha</span>))
<span class="pl-smi">arrowsy</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">res.s4vdpca2</span><span class="pl-k">$</span><span class="pl-smi">v</span><span class="pl-k">*</span>(<span class="pl-smi">res.s4vdpca2</span><span class="pl-k">$</span><span class="pl-smi">d</span><span class="pl-k">^</span>(<span class="pl-c1">1</span><span class="pl-k">-</span><span class="pl-smi">alpha</span>))

<span class="pl-smi">cand</span> <span class="pl-k">&lt;-</span> c(order(abs(<span class="pl-smi">arrowsx</span>))[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>],order(abs(<span class="pl-smi">arrowsy</span>))[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])

<span class="pl-smi">ord</span> <span class="pl-k">&lt;-</span> order(apply(cbind(abs(<span class="pl-smi">arrowsx</span>),abs(<span class="pl-smi">arrowsy</span>)),<span class="pl-c1">1</span>,<span class="pl-smi">max</span>),<span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">T</span>)

<span class="pl-smi">cand</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">20</span>

<span class="pl-smi">p.cutoff.labels</span><span class="pl-k">&lt;-</span>rep(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-c1">4</span>)
<span class="pl-smi">p.cutoff.labels</span>[c(<span class="pl-c1">1</span>,<span class="pl-c1">4</span>,<span class="pl-c1">6</span>,<span class="pl-c1">9</span>)]<span class="pl-k">&lt;-</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">0.01</span>,<span class="pl-c1">0.05</span>,<span class="pl-c1">1</span>)    

<span class="pl-smi">nf</span> <span class="pl-k">&lt;-</span> layout(<span class="pl-k">matrix</span>(c(<span class="pl-c1">2</span>,<span class="pl-c1">4</span>,<span class="pl-c1">1</span>,<span class="pl-c1">3</span>),<span class="pl-c1">2</span>,<span class="pl-c1">2</span>,<span class="pl-v">byrow</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>), c(<span class="pl-c1">2.5</span>,<span class="pl-c1">1.75</span>), c(<span class="pl-c1">1.5</span>,<span class="pl-c1">2.5</span>), <span class="pl-c1">TRUE</span>)
par(<span class="pl-v">mar</span> <span class="pl-k">=</span> c(<span class="pl-c1">4</span>,<span class="pl-c1">4</span>,.<span class="pl-c1">5</span>,.<span class="pl-c1">5</span>)) <span class="pl-c">#(bottom, left, top, right)</span>
plot(<span class="pl-smi">ptx</span>,<span class="pl-smi">pty</span>,<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>,<span class="pl-v">xlim</span><span class="pl-k">=</span>c(<span class="pl-k">-</span><span class="pl-c1">3.25</span>,<span class="pl-c1">3.25</span>),<span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-k">-</span><span class="pl-c1">3.25</span>,<span class="pl-c1">3.25</span>),<span class="pl-v">axes</span><span class="pl-k">=</span><span class="pl-c1">T</span>
,<span class="pl-v">ylab</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">"</span>PC2 /<span class="pl-pds">"</span></span>,sum(<span class="pl-smi">arrowsy</span><span class="pl-k">!=</span><span class="pl-c1">0</span>),<span class="pl-s"><span class="pl-pds">"</span>genes<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>)
     ,<span class="pl-v">xlab</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">"</span>PC1 /<span class="pl-pds">"</span></span>,sum(<span class="pl-smi">arrowsx</span><span class="pl-k">!=</span><span class="pl-c1">0</span>),<span class="pl-s"><span class="pl-pds">"</span>genes<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>),<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
arrows(<span class="pl-c1">0</span>, <span class="pl-c1">0</span>, <span class="pl-v">x1</span> <span class="pl-k">=</span> <span class="pl-smi">arrowsx</span>, <span class="pl-v">y1</span><span class="pl-k">=</span><span class="pl-smi">arrowsy</span>,<span class="pl-v">length</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,<span class="pl-v">col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gray<span class="pl-pds">"</span></span>)
par(<span class="pl-v">xpd</span><span class="pl-k">=</span><span class="pl-c1">NA</span>)
<span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-smi">cand</span>){
  arrows(<span class="pl-c1">0</span>, <span class="pl-c1">0</span>, <span class="pl-v">x1</span> <span class="pl-k">=</span> <span class="pl-smi">arrowsx</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]], <span class="pl-v">y1</span><span class="pl-k">=</span><span class="pl-smi">arrowsy</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]],<span class="pl-v">length</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,<span class="pl-v">col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>)
   text(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">arrowsx</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]]<span class="pl-k">+</span> sign(<span class="pl-smi">arrowsx</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]])<span class="pl-k">*</span>
   .<span class="pl-c1">2</span>,<span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">arrowsy</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]]<span class="pl-k">+</span>sign(<span class="pl-smi">arrowsy</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]])<span class="pl-k">*</span>.<span class="pl-c1">2</span>,
         <span class="pl-v">labels</span><span class="pl-k">=</span><span class="pl-smi">agi2symbol</span>[<span class="pl-smi">ord</span>[<span class="pl-smi">i</span>]],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">cex</span><span class="pl-k">=</span>.<span class="pl-c1">75</span>)
  }
<span class="pl-smi">sec</span> <span class="pl-k">&lt;-</span> which(<span class="pl-smi">cols</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>darkgreen<span class="pl-pds">"</span></span><span class="pl-k">|</span><span class="pl-smi">cols</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>)
points(<span class="pl-smi">ptx</span>,<span class="pl-smi">pty</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">cols</span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-smi">ptype</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">1</span>)
points(<span class="pl-smi">ptx</span>[<span class="pl-smi">sec</span>],<span class="pl-smi">pty</span>[<span class="pl-smi">sec</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">cols</span>[<span class="pl-smi">sec</span>],<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-smi">ptype</span>[<span class="pl-smi">sec</span>],<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">1.25</span>)
par(<span class="pl-v">mar</span> <span class="pl-k">=</span> c(<span class="pl-c1">0</span>,<span class="pl-c1">3</span>,<span class="pl-c1">3</span>,<span class="pl-c1">3</span>)) <span class="pl-c">#(bottom, left, top, right)</span>
V(<span class="pl-smi">g1</span>)<span class="pl-k">$</span><span class="pl-smi">label.cex</span> <span class="pl-k">&lt;-</span> .<span class="pl-c1">85</span>
V(<span class="pl-smi">g2</span>)<span class="pl-k">$</span><span class="pl-smi">label.cex</span> <span class="pl-k">&lt;-</span> .<span class="pl-c1">85</span>
set.seed(<span class="pl-c1">1235</span>)
plot(<span class="pl-smi">g1</span>)
par(<span class="pl-v">mar</span> <span class="pl-k">=</span> c(<span class="pl-c1">0</span>,<span class="pl-c1">3</span>,<span class="pl-c1">0</span>,<span class="pl-c1">3.5</span>)) <span class="pl-c">#(bottom, left, top, right)</span>
set.seed(<span class="pl-c1">1235</span>)
plot(<span class="pl-smi">g2</span>)
plot(c(<span class="pl-k">-</span><span class="pl-c1">1</span>,<span class="pl-c1">1</span>),c(<span class="pl-k">-</span><span class="pl-c1">1</span>,<span class="pl-c1">1</span>),<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>,<span class="pl-v">axes</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
points(
  <span class="pl-v">x</span> <span class="pl-k">=</span> rep(<span class="pl-k">-</span>.<span class="pl-c1">6</span>, <span class="pl-c1">9</span>), 
  <span class="pl-v">y</span> <span class="pl-k">=</span> seq(<span class="pl-k">-</span>.<span class="pl-c1">2</span>, (<span class="pl-k">-</span>.<span class="pl-c1">2</span><span class="pl-k">-</span>(<span class="pl-c1">0.05</span><span class="pl-k">*</span><span class="pl-c1">9</span>)), 
          <span class="pl-v">length.out</span> <span class="pl-k">=</span> <span class="pl-c1">9</span>), 
  <span class="pl-v">pch</span> <span class="pl-k">=</span> <span class="pl-c1">15</span>, <span class="pl-v">col</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>#FF0000<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>#FF1F1F<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>#FF3F3F<span class="pl-pds">"</span></span>,
                    <span class="pl-s"><span class="pl-pds">"</span>#FF5F5F<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>#FF7F7F<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>#FF9F9F<span class="pl-pds">"</span></span>,
                    <span class="pl-s"><span class="pl-pds">"</span>#FFBFBF<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#FFDFDF<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#FFFFFF<span class="pl-pds">"</span></span>)
)

text(
  <span class="pl-v">x</span> <span class="pl-k">=</span> rep(<span class="pl-k">-</span>.<span class="pl-c1">677</span>, <span class="pl-c1">9</span>), 
  <span class="pl-v">y</span> <span class="pl-k">=</span> seq(<span class="pl-k">-</span>.<span class="pl-c1">2</span>, (<span class="pl-k">-</span>.<span class="pl-c1">2</span><span class="pl-k">-</span>(<span class="pl-c1">0.05</span><span class="pl-k">*</span><span class="pl-c1">9</span>)), 
          <span class="pl-v">length.out</span> <span class="pl-k">=</span> <span class="pl-c1">9</span>),
  <span class="pl-v">labels</span> <span class="pl-k">=</span> <span class="pl-smi">p.cutoff.labels</span>, 
  <span class="pl-v">cex</span> <span class="pl-k">=</span> .<span class="pl-c1">7</span>,
  <span class="pl-v">adj</span><span class="pl-k">=</span><span class="pl-c1">1</span>
)
text(
  <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-k">-</span>.<span class="pl-c1">65</span>,
  <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-c1">0.02</span>,
  <span class="pl-v">labels</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Adjusted<span class="pl-cce">\n</span>p-values<span class="pl-pds">"</span></span>,
  <span class="pl-v">cex</span><span class="pl-k">=</span>.<span class="pl-c1">7</span>,
  <span class="pl-v">adj</span><span class="pl-k">=</span> <span class="pl-c1">0.5</span>,
  <span class="pl-v">font</span><span class="pl-k">=</span><span class="pl-c1">2</span>
)

legend(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-c1">0</span>,<span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-k">-</span>.<span class="pl-c1">05</span>,<span class="pl-v">legend</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">'</span>WNT<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>SHH<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>group C<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>group D<span class="pl-pds">'</span></span>)
       ,<span class="pl-v">col</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">'</span>blue<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>darkred<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>orange<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>darkgreen<span class="pl-pds">'</span></span>)
       ,<span class="pl-v">border</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span>c(<span class="pl-c1">18</span>,<span class="pl-c1">20</span>,<span class="pl-c1">15</span>,<span class="pl-c1">17</span>),<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">1</span>,<span class="pl-v">bty</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>)
text(
  <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-c1">0.35</span>,
  <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-k">-</span>.<span class="pl-c1">0225</span>,
  <span class="pl-v">labels</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>subgroup<span class="pl-pds">"</span></span>,
  <span class="pl-v">cex</span><span class="pl-k">=</span>.<span class="pl-c1">7</span>,
  <span class="pl-v">adj</span><span class="pl-k">=</span> <span class="pl-c1">0.5</span>,
  <span class="pl-v">font</span><span class="pl-k">=</span><span class="pl-c1">2</span>
)</pre></div>
</article>
  </div>

</div>

<a href="#jump-to-line" rel="facebox[.linejump]" data-hotkey="l" style="display:none">Jump to Line</a>
<div id="jump-to-line" style="display:none">
  <form accept-charset="UTF-8" class="js-jump-to-line-form">
    <input class="linejump-input js-jump-to-line-field" type="text" placeholder="Jump to line&hellip;" autofocus>
    <button type="submit" class="btn">Go</button>
  </form>
</div>

        </div>

      </div><!-- /.repo-container -->
      <div class="modal-backdrop"></div>
    </div><!-- /.container -->
  </div><!-- /.site -->


    </div><!-- /.wrapper -->

      <div class="container">
  <div class="site-footer" role="contentinfo">
    <ul class="site-footer-links right">
        <li><a href="https://status.github.com/" data-ga-click="Footer, go to status, text:status">Status</a></li>
      <li><a href="https://developer.github.com" data-ga-click="Footer, go to api, text:api">API</a></li>
      <li><a href="https://training.github.com" data-ga-click="Footer, go to training, text:training">Training</a></li>
      <li><a href="https://shop.github.com" data-ga-click="Footer, go to shop, text:shop">Shop</a></li>
        <li><a href="https://github.com/blog" data-ga-click="Footer, go to blog, text:blog">Blog</a></li>
        <li><a href="https://github.com/about" data-ga-click="Footer, go to about, text:about">About</a></li>

    </ul>

    <a href="https://github.com" aria-label="Homepage">
      <span class="mega-octicon octicon-mark-github" title="GitHub"></span>
</a>
    <ul class="site-footer-links">
      <li>&copy; 2015 <span title="0.03525s from github-fe139-cp1-prd.iad.github.net">GitHub</span>, Inc.</li>
        <li><a href="https://github.com/site/terms" data-ga-click="Footer, go to terms, text:terms">Terms</a></li>
        <li><a href="https://github.com/site/privacy" data-ga-click="Footer, go to privacy, text:privacy">Privacy</a></li>
        <li><a href="https://github.com/security" data-ga-click="Footer, go to security, text:security">Security</a></li>
        <li><a href="https://github.com/contact" data-ga-click="Footer, go to contact, text:contact">Contact</a></li>
    </ul>
  </div>
</div>


    <div class="fullscreen-overlay js-fullscreen-overlay" id="fullscreen_overlay">
  <div class="fullscreen-container js-suggester-container">
    <div class="textarea-wrap">
      <textarea name="fullscreen-contents" id="fullscreen-contents" class="fullscreen-contents js-fullscreen-contents" placeholder=""></textarea>
      <div class="suggester-container">
        <div class="suggester fullscreen-suggester js-suggester js-navigation-container"></div>
      </div>
    </div>
  </div>
  <div class="fullscreen-sidebar">
    <a href="#" class="exit-fullscreen js-exit-fullscreen tooltipped tooltipped-w" aria-label="Exit Zen Mode">
      <span class="mega-octicon octicon-screen-normal"></span>
    </a>
    <a href="#" class="theme-switcher js-theme-switcher tooltipped tooltipped-w"
      aria-label="Switch themes">
      <span class="octicon octicon-color-mode"></span>
    </a>
  </div>
</div>



    
    

    <div id="ajax-error-message" class="flash flash-error">
      <span class="octicon octicon-alert"></span>
      <a href="#" class="octicon octicon-x flash-close js-ajax-error-dismiss" aria-label="Dismiss error"></a>
      Something went wrong with that request. Please try again.
    </div>


      <script crossorigin="anonymous" src="https://assets-cdn.github.com/assets/frameworks-d22b59d0085e83b7549ba4341ec9e68f80c2f29c8e49213ee182003dc8d568c6.js"></script>
      <script async="async" crossorigin="anonymous" src="https://assets-cdn.github.com/assets/github-b1799c46bb59be68d925cba885ab38303711632ad670e579e1bc4857963e52cb.js"></script>
      
      

  </body>
</html>

