


<!DOCTYPE html>
<html lang="en" class="">
  <head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# object: http://ogp.me/ns/object# article: http://ogp.me/ns/article# profile: http://ogp.me/ns/profile#">
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="Content-Language" content="en">
    
    
    <title>NELSI/README.md at master · sebastianduchene/NELSI · GitHub</title>
    <link rel="search" type="application/opensearchdescription+xml" href="/opensearch.xml" title="GitHub">
    <link rel="fluid-icon" href="https://github.com/fluidicon.png" title="GitHub">
    <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-114.png">
    <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114.png">
    <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-144.png">
    <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144.png">
    <meta property="fb:app_id" content="1401488693436528">

      <meta content="@github" name="twitter:site" /><meta content="summary" name="twitter:card" /><meta content="sebastianduchene/NELSI" name="twitter:title" /><meta content="This is the repository for NELSI: Nucleotide Evolutionary rate Simulator" name="twitter:description" /><meta content="https://avatars1.githubusercontent.com/u/6167278?v=3&amp;s=400" name="twitter:image:src" />
      <meta content="GitHub" property="og:site_name" /><meta content="object" property="og:type" /><meta content="https://avatars1.githubusercontent.com/u/6167278?v=3&amp;s=400" property="og:image" /><meta content="sebastianduchene/NELSI" property="og:title" /><meta content="https://github.com/sebastianduchene/NELSI" property="og:url" /><meta content="This is the repository for NELSI: Nucleotide Evolutionary rate Simulator" property="og:description" />
      <meta name="browser-stats-url" content="/_stats">
    <link rel="assets" href="https://assets-cdn.github.com/">
    
    <meta name="pjax-timeout" content="1000">
    

    <meta name="msapplication-TileImage" content="/windows-tile.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="selected-link" value="repo_source" data-pjax-transient>
      <meta name="google-analytics" content="UA-3769691-2">

    <meta content="collector.githubapp.com" name="octolytics-host" /><meta content="collector-cdn.github.com" name="octolytics-script-host" /><meta content="github" name="octolytics-app-id" /><meta content="C1BEC102:4D96:725B6C8:55103FA7" name="octolytics-dimension-request_id" />
    
    <meta content="Rails, view, blob#show" name="analytics-event" />

    
    <link rel="icon" type="image/x-icon" href="https://assets-cdn.github.com/favicon.ico">


    <meta content="authenticity_token" name="csrf-param" />
<meta content="A4E5xzio2Qj82Gmsd8YMLoG7FiRysDjD5EhCjLYJPQBBBLGaVTV824LFVlkwoig7SzFPC/ZEbhK69huewtEZPQ==" name="csrf-token" />

    <link href="https://assets-cdn.github.com/assets/github-099e0ecc2851c8aca89ef6dafa191df3b0f2a2c8ad34e134c5473ca1ba0a59b2.css" media="all" rel="stylesheet" />
    <link href="https://assets-cdn.github.com/assets/github2-1171344316fc088255ee2a06c271d14240f1a4e06985fe9e897762947872e858.css" media="all" rel="stylesheet" />
    
    


    <meta http-equiv="x-pjax-version" content="c0f32272c66bfb10ed7d46b7c88c6299">

      
  <meta name="description" content="This is the repository for NELSI: Nucleotide Evolutionary rate Simulator">
  <meta name="go-import" content="github.com/sebastianduchene/NELSI git https://github.com/sebastianduchene/NELSI.git">

  <meta content="6167278" name="octolytics-dimension-user_id" /><meta content="sebastianduchene" name="octolytics-dimension-user_login" /><meta content="17787440" name="octolytics-dimension-repository_id" /><meta content="sebastianduchene/NELSI" name="octolytics-dimension-repository_nwo" /><meta content="true" name="octolytics-dimension-repository_public" /><meta content="false" name="octolytics-dimension-repository_is_fork" /><meta content="17787440" name="octolytics-dimension-repository_network_root_id" /><meta content="sebastianduchene/NELSI" name="octolytics-dimension-repository_network_root_nwo" />
  <link href="https://github.com/sebastianduchene/NELSI/commits/master.atom" rel="alternate" title="Recent Commits to NELSI:master" type="application/atom+xml">

  </head>


  <body class="logged_out  env-production  vis-public page-blob">
    <a href="#start-of-content" tabindex="1" class="accessibility-aid js-skip-to-content">Skip to content</a>
    <div class="wrapper">
      
      
      


        
        <div class="header header-logged-out" role="banner">
  <div class="container clearfix">

    <a class="header-logo-wordmark" href="https://github.com/" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark">
      <span class="mega-octicon octicon-logo-github"></span>
    </a>

    <div class="header-actions" role="navigation">
        <a class="btn btn-primary" href="/join" data-ga-click="(Logged out) Header, clicked Sign up, text:sign-up">Sign up</a>
      <a class="btn" href="/login?return_to=%2Fsebastianduchene%2FNELSI%2Fblob%2Fmaster%2FREADME.md" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in">Sign in</a>
    </div>

    <div class="site-search repo-scope js-site-search" role="search">
      <form accept-charset="UTF-8" action="/sebastianduchene/NELSI/search" class="js-site-search-form" data-global-search-url="/search" data-repo-search-url="/sebastianduchene/NELSI/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
  <input type="text"
    class="js-site-search-field is-clearable"
    data-hotkey="s"
    name="q"
    placeholder="Search"
    data-global-scope-placeholder="Search GitHub"
    data-repo-scope-placeholder="Search"
    tabindex="1"
    autocapitalize="off">
  <div class="scope-badge">This repository</div>
</form>
    </div>

      <ul class="header-nav left" role="navigation">
          <li class="header-nav-item">
            <a class="header-nav-link" href="/explore" data-ga-click="(Logged out) Header, go to explore, text:explore">Explore</a>
          </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="/features" data-ga-click="(Logged out) Header, go to features, text:features">Features</a>
          </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="https://enterprise.github.com/" data-ga-click="(Logged out) Header, go to enterprise, text:enterprise">Enterprise</a>
          </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="/blog" data-ga-click="(Logged out) Header, go to blog, text:blog">Blog</a>
          </li>
      </ul>

  </div>
</div>



      <div id="start-of-content" class="accessibility-aid"></div>
          <div class="site" itemscope itemtype="http://schema.org/WebPage">
    <div id="js-flash-container">
      
    </div>
    <div class="pagehead repohead instapaper_ignore readability-menu">
      <div class="container">
        
<ul class="pagehead-actions">

  <li>
      <a href="/login?return_to=%2Fsebastianduchene%2FNELSI"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to watch a repository" rel="nofollow">
    <span class="octicon octicon-eye"></span>
    Watch
  </a>
  <a class="social-count" href="/sebastianduchene/NELSI/watchers">
    2
  </a>

  </li>

  <li>
      <a href="/login?return_to=%2Fsebastianduchene%2FNELSI"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to star a repository" rel="nofollow">
    <span class="octicon octicon-star"></span>
    Star
  </a>

    <a class="social-count js-social-count" href="/sebastianduchene/NELSI/stargazers">
      2
    </a>

  </li>

    <li>
      <a href="/login?return_to=%2Fsebastianduchene%2FNELSI"
        class="btn btn-sm btn-with-count tooltipped tooltipped-n"
        aria-label="You must be signed in to fork a repository" rel="nofollow">
        <span class="octicon octicon-repo-forked"></span>
        Fork
      </a>
      <a href="/sebastianduchene/NELSI/network" class="social-count">
        1
      </a>
    </li>
</ul>

        <h1 itemscope itemtype="http://data-vocabulary.org/Breadcrumb" class="entry-title public">
          <span class="mega-octicon octicon-repo"></span>
          <span class="author"><a href="/sebastianduchene" class="url fn" itemprop="url" rel="author"><span itemprop="title">sebastianduchene</span></a></span><!--
       --><span class="path-divider">/</span><!--
       --><strong><a href="/sebastianduchene/NELSI" class="js-current-repository" data-pjax="#js-repo-pjax-container">NELSI</a></strong>

          <span class="page-context-loader">
            <img alt="" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
          </span>

        </h1>
      </div><!-- /.container -->
    </div><!-- /.repohead -->

    <div class="container">
      <div class="repository-with-sidebar repo-container new-discussion-timeline  ">
        <div class="repository-sidebar clearfix">
            
<nav class="sunken-menu repo-nav js-repo-nav js-sidenav-container-pjax js-octicon-loaders"
     role="navigation"
     data-pjax="#js-repo-pjax-container"
     data-issue-count-url="/sebastianduchene/NELSI/issues/counts">
  <ul class="sunken-menu-group">
    <li class="tooltipped tooltipped-w" aria-label="Code">
      <a href="/sebastianduchene/NELSI" aria-label="Code" class="selected js-selected-navigation-item sunken-menu-item" data-hotkey="g c" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches /sebastianduchene/NELSI">
        <span class="octicon octicon-code"></span> <span class="full-word">Code</span>
        <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>

      <li class="tooltipped tooltipped-w" aria-label="Issues">
        <a href="/sebastianduchene/NELSI/issues" aria-label="Issues" class="js-selected-navigation-item sunken-menu-item" data-hotkey="g i" data-selected-links="repo_issues repo_labels repo_milestones /sebastianduchene/NELSI/issues">
          <span class="octicon octicon-issue-opened"></span> <span class="full-word">Issues</span>
          <span class="js-issue-replace-counter"></span>
          <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>      </li>

    <li class="tooltipped tooltipped-w" aria-label="Pull requests">
      <a href="/sebastianduchene/NELSI/pulls" aria-label="Pull requests" class="js-selected-navigation-item sunken-menu-item" data-hotkey="g p" data-selected-links="repo_pulls /sebastianduchene/NELSI/pulls">
          <span class="octicon octicon-git-pull-request"></span> <span class="full-word">Pull requests</span>
          <span class="js-pull-replace-counter"></span>
          <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>


  </ul>
  <div class="sunken-menu-separator"></div>
  <ul class="sunken-menu-group">

    <li class="tooltipped tooltipped-w" aria-label="Pulse">
      <a href="/sebastianduchene/NELSI/pulse" aria-label="Pulse" class="js-selected-navigation-item sunken-menu-item" data-selected-links="pulse /sebastianduchene/NELSI/pulse">
        <span class="octicon octicon-pulse"></span> <span class="full-word">Pulse</span>
        <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>

    <li class="tooltipped tooltipped-w" aria-label="Graphs">
      <a href="/sebastianduchene/NELSI/graphs" aria-label="Graphs" class="js-selected-navigation-item sunken-menu-item" data-selected-links="repo_graphs repo_contributors /sebastianduchene/NELSI/graphs">
        <span class="octicon octicon-graph"></span> <span class="full-word">Graphs</span>
        <img alt="" class="mini-loader" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-e513294efa576953719e4e2de888dd9cf929b7d62ed8d05f25e731d02452ab6c.gif" width="16" />
</a>    </li>
  </ul>


</nav>

              <div class="only-with-full-nav">
                  
<div class="clone-url open"
  data-protocol-type="http"
  data-url="/users/set_protocol?protocol_selector=http&amp;protocol_type=clone">
  <h3><span class="text-emphasized">HTTPS</span> clone URL</h3>
  <div class="input-group js-zeroclipboard-container">
    <input type="text" class="input-mini input-monospace js-url-field js-zeroclipboard-target"
           value="https://github.com/sebastianduchene/NELSI.git" readonly="readonly">
    <span class="input-group-button">
      <button aria-label="Copy to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button" data-copied-hint="Copied!" type="button"><span class="octicon octicon-clippy"></span></button>
    </span>
  </div>
</div>

  
<div class="clone-url "
  data-protocol-type="subversion"
  data-url="/users/set_protocol?protocol_selector=subversion&amp;protocol_type=clone">
  <h3><span class="text-emphasized">Subversion</span> checkout URL</h3>
  <div class="input-group js-zeroclipboard-container">
    <input type="text" class="input-mini input-monospace js-url-field js-zeroclipboard-target"
           value="https://github.com/sebastianduchene/NELSI" readonly="readonly">
    <span class="input-group-button">
      <button aria-label="Copy to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button" data-copied-hint="Copied!" type="button"><span class="octicon octicon-clippy"></span></button>
    </span>
  </div>
</div>



<p class="clone-options">You can clone with
  <a href="#" class="js-clone-selector" data-protocol="http">HTTPS</a> or <a href="#" class="js-clone-selector" data-protocol="subversion">Subversion</a>.
  <a href="https://help.github.com/articles/which-remote-url-should-i-use" class="help tooltipped tooltipped-n" aria-label="Get help on which URL is right for you.">
    <span class="octicon octicon-question"></span>
  </a>
</p>



                <a href="/sebastianduchene/NELSI/archive/master.zip"
                   class="btn btn-sm sidebar-button"
                   aria-label="Download the contents of sebastianduchene/NELSI as a zip file"
                   title="Download the contents of sebastianduchene/NELSI as a zip file"
                   rel="nofollow">
                  <span class="octicon octicon-cloud-download"></span>
                  Download ZIP
                </a>
              </div>
        </div><!-- /.repository-sidebar -->

        <div id="js-repo-pjax-container" class="repository-content context-loader-container" data-pjax-container>
          

<a href="/sebastianduchene/NELSI/blob/45d720da3dfd8a7e08d25fb35d3e97c3a5c36e44/README.md" class="hidden js-permalink-shortcut" data-hotkey="y">Permalink</a>

<!-- blob contrib key: blob_contributors:v21:10b49a5f5bfabe2d326eb04d8b9c07c7 -->

<div class="file-navigation js-zeroclipboard-container">
  
<div class="select-menu js-menu-container js-select-menu left">
  <span class="btn btn-sm select-menu-button js-menu-target css-truncate" data-hotkey="w"
    data-master-branch="master"
    data-ref="master"
    title="master"
    role="button" aria-label="Switch branches or tags" tabindex="0" aria-haspopup="true">
    <span class="octicon octicon-git-branch"></span>
    <i>branch:</i>
    <span class="js-select-button css-truncate-target">master</span>
  </span>

  <div class="select-menu-modal-holder js-menu-content js-navigation-container" data-pjax aria-hidden="true">

    <div class="select-menu-modal">
      <div class="select-menu-header">
        <span class="select-menu-title">Switch branches/tags</span>
        <span class="octicon octicon-x js-menu-close" role="button" aria-label="Close"></span>
      </div>

      <div class="select-menu-filters">
        <div class="select-menu-text-filter">
          <input type="text" aria-label="Filter branches/tags" id="context-commitish-filter-field" class="js-filterable-field js-navigation-enable" placeholder="Filter branches/tags">
        </div>
        <div class="select-menu-tabs">
          <ul>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="branches" data-filter-placeholder="Filter branches/tags" class="js-select-menu-tab">Branches</a>
            </li>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="tags" data-filter-placeholder="Find a tag…" class="js-select-menu-tab">Tags</a>
            </li>
          </ul>
        </div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="branches">

        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


            <a class="select-menu-item js-navigation-item js-navigation-open selected"
               href="/sebastianduchene/NELSI/blob/master/README.md"
               data-name="master"
               data-skip-pjax="true"
               rel="nofollow">
              <span class="select-menu-item-icon octicon octicon-check"></span>
              <span class="select-menu-item-text css-truncate-target" title="master">
                master
              </span>
            </a>
            <a class="select-menu-item js-navigation-item js-navigation-open "
               href="/sebastianduchene/NELSI/blob/testing/README.md"
               data-name="testing"
               data-skip-pjax="true"
               rel="nofollow">
              <span class="select-menu-item-icon octicon octicon-check"></span>
              <span class="select-menu-item-text css-truncate-target" title="testing">
                testing
              </span>
            </a>
            <a class="select-menu-item js-navigation-item js-navigation-open "
               href="/sebastianduchene/NELSI/blob/testing_d/README.md"
               data-name="testing_d"
               data-skip-pjax="true"
               rel="nofollow">
              <span class="select-menu-item-icon octicon octicon-check"></span>
              <span class="select-menu-item-text css-truncate-target" title="testing_d">
                testing_d
              </span>
            </a>
        </div>

          <div class="select-menu-no-results">Nothing to show</div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="tags">
        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


        </div>

        <div class="select-menu-no-results">Nothing to show</div>
      </div>

    </div>
  </div>
</div>

  <div class="btn-group right">
    <a href="/sebastianduchene/NELSI/find/master"
          class="js-show-file-finder btn btn-sm empty-icon tooltipped tooltipped-s"
          data-pjax
          data-hotkey="t"
          aria-label="Quickly jump between files">
      <span class="octicon octicon-list-unordered"></span>
    </a>
    <button aria-label="Copy file path to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button" data-copied-hint="Copied!" type="button"><span class="octicon octicon-clippy"></span></button>
  </div>

  <div class="breadcrumb js-zeroclipboard-target">
    <span class='repo-root js-repo-root'><span itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a href="/sebastianduchene/NELSI" class="" data-branch="master" data-direction="back" data-pjax="true" itemscope="url"><span itemprop="title">NELSI</span></a></span></span><span class="separator">/</span><strong class="final-path">README.md</strong>
  </div>
</div>

<include-fragment class="commit commit-loader file-history-tease" src="/sebastianduchene/NELSI/contributors/master/README.md">
  <div class="file-history-tease-header">
    Fetching contributors&hellip;
  </div>

  <div class="participation">
    <p class="loader-loading"><img alt="" height="16" src="https://assets-cdn.github.com/assets/spinners/octocat-spinner-32-EAF2F5-0bdc57d34b85c4a4de9d0d1db10cd70e8a95f33ff4f46c5a8c48b4bf4e5a9abe.gif" width="16" /></p>
    <p class="loader-error">Cannot retrieve contributors at this time</p>
  </div>
</include-fragment>
<div class="file">
  <div class="file-header">
    <div class="file-actions">

      <div class="btn-group">
        <a href="/sebastianduchene/NELSI/raw/master/README.md" class="btn btn-sm " id="raw-url">Raw</a>
          <a href="/sebastianduchene/NELSI/blame/master/README.md" class="btn btn-sm js-update-url-with-hash">Blame</a>
        <a href="/sebastianduchene/NELSI/commits/master/README.md" class="btn btn-sm " rel="nofollow">History</a>
      </div>


          <button type="button" class="octicon-btn disabled tooltipped tooltipped-n" aria-label="You must be signed in to make or propose changes">
            <span class="octicon octicon-pencil"></span>
          </button>

        <button type="button" class="octicon-btn octicon-btn-danger disabled tooltipped tooltipped-n" aria-label="You must be signed in to make or propose changes">
          <span class="octicon octicon-trashcan"></span>
        </button>
    </div>

    <div class="file-info">
        416 lines (241 sloc)
        <span class="file-info-divider"></span>
      19.774 kb
    </div>
  </div>
    <div id="readme" class="blob instapaper_body">
    <article class="markdown-body entry-content" itemprop="mainContentOfPage"><h1>
<a id="user-content-nelsi-nucleotide-evolutionary-simulator" class="anchor" href="#nelsi-nucleotide-evolutionary-simulator" aria-hidden="true"><span class="octicon octicon-link"></span></a>NELSI: Nucleotide EvoLutionary Simulator</h1>

<p>David Duchene and Sebastian Duchene</p>

<p>david.duchene[at]anu.edu.au</p>

<p>sebastian.duchene[at]sydney.edu.au</p>

<p>15 December 2014</p>

<h2>
<a id="user-content-new-features" class="anchor" href="#new-features" aria-hidden="true"><span class="octicon octicon-link"></span></a>New features</h2>

<ul class="task-list">
<li><p>New function to simulate rates under the dirichlet process, as described by <a href="#references">Heath <em>et al</em>. 2012</a>.</p></li>
<li><p>New function to obtain node times for internal nodes only. Can be used for ultrametric and non-ultrametric trees.</p></li>
<li><p>Experimental function to get all the tips that descend from a node. Similar to function 'tips' from GEIGER.</p></li>
</ul>

<h2>
<a id="user-content-introduction" class="anchor" href="#introduction" aria-hidden="true"><span class="octicon octicon-link"></span></a>Introduction</h2>

<p>Models for molecular rate variation play a key role in molecular phylogenetics. Due to their importance in evolutionary biology, there is a wide variety of models, which can be classified into five broad categories: </p>

<ul class="task-list">
<li><p>Strict clock, where a single rate is assumed for all the branches of a phylogeny, as described by <a href="#references">Zuckerkandl, E. and Pauling, L. (1962)</a>. </p></li>
<li><p>Local clock, in which there is a fixed number of strict clocks in a phylogenetic tree, so that the rate is constant for some lineages (<a href="#references">Hasegawa <em>et al</em>. 1989</a>, <a href="#references">Yoder and Yang. 2000</a>).</p></li>
<li><p>Autocorrelated models, where rates along a lineage evolve gradually, and therefore display a degree of correlation, such as that described by <a href="#references">Thorne <em>et al</em>. (1998)</a>. </p></li>
<li><p>Uncorrelated models, where the rates for each of the branches are independently and identically distributed variables, drawn from a specified probability function, sucha as the uncorrelated lognormal clock (<a href="#references">Drummond <em>et al</em>. 2006</a>).</p></li>
<li><p>Unconstrained models, where the rates for all branches are independent and drawn from different distributions (<a href="#references">Lepage <em>et al</em> 2007</a>). </p></li>
</ul>

<p>With the increasing development of clock models, it is necessary to assess their performance with simulations and analyses of empirical data. We present NELSI, an R package to simulate rates of evolution along phylogenetic trees. The principle is similar to the program RateEvolver (<a href="#references">Ho <em>et al</em>. 2005</a>), but it allows more flexibility and it can be easily combined with popular programs to simulate phylogenetic trees. In the current version we have implemented some of the most popular methods, but the pacakage is under constant development and we will include more models, as they become available. For requests and reporting bugs please contact sebastian.duchene[at]sydney.edu.au.</p>

<h2>
<a id="user-content-description" class="anchor" href="#description" aria-hidden="true"><span class="octicon octicon-link"></span></a>Description</h2>

<p>NELSI is implemented in the R programming language, and it is available as a package in <a href="https://github.com/sebastianduchene/NELSI/">github</a>. It is compatible with some popular phylogenetic packages in R, such as Ape (<a href="#references">Popescu <em>et al</em>. 2012</a>) and phangorn (<a href="#references">Schliep 2011</a>) making it accessible to users familiar with phylogenetic data in R. The main functions use phylogenetic trees of class phylo, with branch lengths representing units of time. Trees estimated in other programs can be imported with ape in NEWICK or NEXUS format. Some R packages that simulate phylogenetic trees, such as geiger and TreeSim, also produce trees of class phylo, which can be used directly with NELSI. An important requirement of simulating rates of evolution along phylogenetic trees is that the trees should correspond to chronograms, with branch lengths in units of time. </p>

<h1>
<a id="user-content-tutorial" class="anchor" href="#tutorial" aria-hidden="true"><span class="octicon octicon-link"></span></a>Tutorial</h1>

<h2>
<a id="user-content-1-installation-and-setup" class="anchor" href="#1-installation-and-setup" aria-hidden="true"><span class="octicon octicon-link"></span></a>1. Installation and setup</h2>

<p>NELSI requires a recent version of R (&gt;=2.5). If R is not installed in your machine, download and install the appropriate version <a href="/sebastianduchene/NELSI/blob/master/www.r-project.org">here</a>. </p>

<p>R packages can be downloaded and installed directly from <a href="/sebastianduchene/NELSI/blob/master/github.com/sebastianduchene/nelsi">github</a> with the package <a href="https://github.com/hadley/devtools">devtools</a>. This is the easiest way to install NELSI (and many other packages). </p>

<p>We will begin by installing devtools from the Comprehensive R Archive Network (CRAN). Please follow the instructions bellow:</p>

<ul class="task-list">
<li><p>Open the R console by clicking on the R icon in your desktop or in Applications (depending on the operating system)</p></li>
<li><p>Make sure that you have an internet connection and type in the code bellow:</p></li>
</ul>

<div class="highlight highlight-coffee"><pre>install.packages(<span class="pl-s"><span class="pl-pds">"</span>devtools<span class="pl-pds">"</span></span>)</pre></div>

<p>Follow the instructions in the prompt. </p>

<ul class="task-list">
<li>Load devtools with the following code:</li>
</ul>

<div class="highlight highlight-coffee"><pre>library(devtools)</pre></div>

<ul class="task-list">
<li>The devtools package has a function to download packages from github repositories. To download and install NELSI type the following at the prompt:</li>
</ul>

<div class="highlight highlight-coffee"><pre>install_github(<span class="pl-v"><span class="pl-v">rep <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>NELSI<span class="pl-pds">"</span></span>, <span class="pl-v"><span class="pl-v">username <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>sebastianduchene<span class="pl-pds">"</span></span>)</pre></div>

<ul class="task-list">
<li>NELSI is now installed. To make all the functions available, load the package by typing:</li>
</ul>

<div class="highlight highlight-coffee"><pre>library(NELSI)</pre></div>

<p>This is all for the installation of NELSI. Please contact the authors to report any bugs. </p>

<p>In the next sections of this tutorial we show an overview of some of the functions available. For a more comprehensive list, please see the manual by typing the following code in the R console:</p>

<div class="highlight highlight-coffee"><pre>help(<span class="pl-k">package</span> <span class="pl-k">=</span> NELSI)</pre></div>

<p>Some examples used to generate the data for the study by  Ho. <em>et al</em> (<em>In Prep</em>) are available <a href="https://github.com/sebastianduchene/NELSI/tree/master/Ho_et_al_examples">here</a>.</p>

<h2>
<a id="user-content-2-loading-phylogenetic-trees" class="anchor" href="#2-loading-phylogenetic-trees" aria-hidden="true"><span class="octicon octicon-link"></span></a>2. Loading phylogenetic trees</h2>

<p>To simulate rates of evolution we need a phylogenetic tree in which the branch lengths repesetn units of time, known as a chronogram. We can simulate this kind of tree in R, but for this tutorial we will load the tree in the example_data folder in <a href="/sebastianduchene/NELSI/blob/master/github.com/sebastianduchene/nelsi">github</a>. </p>

<ul class="task-list">
<li>Set the R <a href="http://www.statmethods.net/interface/workspace.html">working directory</a> to the example_data folder. Load the example tree with the following code:</li>
</ul>

<div class="highlight highlight-coffee"><pre>myTree <span class="pl-k">&lt;-</span> read.tree(<span class="pl-s"><span class="pl-pds">"</span>tr_example.tree<span class="pl-pds">"</span></span>)</pre></div>

<ul class="task-list">
<li>To get more insight into the chronogram that we have loaded, we can plot it and annotate each node with its age.</li>
</ul>

<div class="highlight highlight-coffee"><pre>plot(myTree)
node.ages <span class="pl-k">&lt;-</span> round(branching.times(myTree), <span class="pl-c1">2</span>)
nodelabels(node.ages, <span class="pl-v"><span class="pl-v">bg <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>)</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-4.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-4.png" alt="plot of chunk unnamed-chunk-4" style="max-width:100%;"></a> </p>

<h2>
<a id="user-content-3-simulate-constant-rates-through-time-strict-clock" class="anchor" href="#3-simulate-constant-rates-through-time-strict-clock" aria-hidden="true"><span class="octicon octicon-link"></span></a>3. Simulate constant rates through time (strict clock)</h2>

<p>The simplest rate simulation model in NELSI is a strict clock, where every branch is given the same rate, with a user-specified noise level. To simulate rates under this model for our chronogram we use the function simulate.cock, which receives as arguments the chronogram, and two parameters: the mean rate and the amount of noise. </p>

<ul class="task-list">
<li>As an example, we will simulate a high rate of substitutions with no noise.</li>
</ul>

<div class="highlight highlight-coffee"><pre>clock.sim <span class="pl-k">&lt;-</span> simulate.clock(myTree, <span class="pl-v"><span class="pl-v">params <span class="pl-k">=</span></span></span> list(<span class="pl-v"><span class="pl-v">rate <span class="pl-k">=</span></span></span> <span class="pl-c1">0.03</span>, <span class="pl-v"><span class="pl-v">noise <span class="pl-k">=</span></span></span> <span class="pl-c1">0</span>))
</pre></div>

<p>The variable clock.sim is an object of class ratesim, which is the output of all the rate simulation functions in NELSI. ratesim objects have two elements. The first is a phylogram (our input topology but with branch lengths in terms of substitutions). The second element in an object of class ratesim is a tree.data.matrix, which is a matrix with all the data about a phylogeny, includng the simulated data. The columns of a tree.data.matrix are the following: (1) is the index of each branch; (2) and (3) are the edge attribute of the class phylo, showing the parent and daughter nodes for each branchrespectively; (4) is the mid age of each branch; (5) is the simulated molecular rate for every branch; (6) is the branch lengths in substitutions per site; and (7) is branch lengths in time units.</p>

<ul class="task-list">
<li>Inspect the tree data matrix as shown bellow:</li>
</ul>

<div class="highlight highlight-coffee"><pre>clock.sim$tree.data.matrix

<span class="pl-c">#      branch.index parent.node daughter.node branch.midage branch.rate</span>
<span class="pl-c"># [1,]            1          11            12   0.325090501        0.03</span>
<span class="pl-c"># [2,]            2          12            13   0.080306199        0.03</span>
<span class="pl-c"># [3,]            3          13             1   0.009929729        0.03</span>
<span class="pl-c"># [4,]            4          13             2   0.009929729        0.03</span>
<span class="pl-c"># [5,]            5          12             3   0.070376470        0.03</span>
<span class="pl-c"># [6,]            6          11            14   0.440591467        0.03</span></pre></div>

<ul class="task-list">
<li>To observe how the rate changes through time in each lineage, you can plot the output of your simulation function directly using the ratesim object. The fist plot will show the rate through time for each lineage, while the second shows the chronogram with the tips coloured proportional to the rate. Therefore, colours of lines in the first plot correspond to the colours of tips in the second plot. The width of the branches is proportional to the rate. With the strict clock there is no rate variation among lineages.</li>
</ul>

<div class="highlight highlight-coffee"><pre>plot(clock.sim, <span class="pl-v"><span class="pl-v">col.lineages <span class="pl-k">=</span></span></span> rainbow(<span class="pl-c1">20</span>), <span class="pl-v"><span class="pl-v">type <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>s<span class="pl-pds">"</span></span>)</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-6.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-6.png" alt="plot of chunk unnamed-chunk-6" style="max-width:100%;"></a> </p>

<h2>
<a id="user-content-4-simulate-autocorrelated-rates" class="anchor" href="#4-simulate-autocorrelated-rates" aria-hidden="true"><span class="octicon octicon-link"></span></a>4. Simulate autocorrelated rates</h2>

<p>One way to relax the assumption of having a single rate throughout is to propose small changes in rate from one branch to the next. The functions simulate.autocor.kishino and simulate.autocor.thorne use the method described in <a href="#references">Kishino <em>et al</em>.(2001)</a> and <a href="#references">Thorne <em>et al.</em>(1998)</a> methods to simulate this kind of rate pattern. In both functions the user only needs to provide the rate at the root of the phylogeny and the amount of autocorrelation, given by the parameter v. </p>

<ul class="task-list">
<li>Using the following code simulate and plot autocorrelated rates using simulate.autocor.kishino; first with low autocorrelation, and then with high autocorrelation.</li>
</ul>

<div class="highlight highlight-coffee"><pre>sim.low.autocor <span class="pl-k">&lt;-</span> simulate.autocor.kishino(myTree, <span class="pl-v"><span class="pl-v">params <span class="pl-k">=</span></span></span> list(<span class="pl-v"><span class="pl-v">initial.rate <span class="pl-k">=</span></span></span> <span class="pl-c1">0.01</span>, 
    <span class="pl-v"><span class="pl-v">v <span class="pl-k">=</span></span></span> <span class="pl-c1">0.1</span>))
sim.high.autocor <span class="pl-k">&lt;-</span> simulate.autocor.kishino(myTree, <span class="pl-v"><span class="pl-v">params <span class="pl-k">=</span></span></span> list(<span class="pl-v"><span class="pl-v">initial.rate <span class="pl-k">=</span></span></span> <span class="pl-c1">0.01</span>, 
    <span class="pl-v"><span class="pl-v">v <span class="pl-k">=</span></span></span> <span class="pl-c1">0.006</span>))
plot(sim.low.autocor, <span class="pl-v"><span class="pl-v">col.lineages <span class="pl-k">=</span></span></span> rainbow(<span class="pl-c1">20</span>), <span class="pl-v"><span class="pl-v">type <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>s<span class="pl-pds">"</span></span>)</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-7.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-7.png" alt="plot of chunk unnamed-chunk-7" style="max-width:100%;"></a> </p>

<div class="highlight highlight-coffee"><pre>plot(sim.high.autocor, <span class="pl-v"><span class="pl-v">col.lineages <span class="pl-k">=</span></span></span> rainbow(<span class="pl-c1">20</span>), <span class="pl-v"><span class="pl-v">type <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>s<span class="pl-pds">"</span></span>)</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-8.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-8.png" alt="plot of chunk unnamed-chunk-8" style="max-width:100%;"></a> </p>

<h2>
<a id="user-content-5-simulate-uncorrelated-lognormal-rates" class="anchor" href="#5-simulate-uncorrelated-lognormal-rates" aria-hidden="true"><span class="octicon octicon-link"></span></a>5. Simulate uncorrelated lognormal rates</h2>

<p>To simulate rates that are uncorrelated among branches, but are independently and identically drawn from a parent distribution, we have implemented three different models for rate simulation. Each function requires different input parameters, as described in <a href="#references">Drummond <em>et al.</em> (2006)</a>.</p>

<ul class="task-list">
<li>Using the following you can simulate rates under an uncorrelated lognormal rates model, which requires the log mean and the standard deviation of the parent distribution. Note that the width of the branches varies, representing rate variation among the branches.</li>
</ul>

<div class="highlight highlight-coffee"><pre>sim.uncor <span class="pl-k">&lt;-</span> simulate.uncor.lnorm(myTree, <span class="pl-v"><span class="pl-v">params <span class="pl-k">=</span></span></span> list(<span class="pl-v"><span class="pl-v">mean.log <span class="pl-k">=</span></span></span> <span class="pl-k">-</span><span class="pl-c1">3.9</span>, <span class="pl-v"><span class="pl-v">sd.log <span class="pl-k">=</span></span></span> <span class="pl-c1">0.7</span>))
plot(sim.uncor, <span class="pl-v"><span class="pl-v">col.lineages <span class="pl-k">=</span></span></span> rainbow(<span class="pl-c1">20</span>), <span class="pl-v"><span class="pl-v">type <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>s<span class="pl-pds">"</span></span>)</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-9.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-9.png" alt="plot of chunk unnamed-chunk-9" style="max-width:100%;"></a> </p>

<p>There are other methods for rate simulation in NELSI, but this tutorial covers the most well-known models. Please refer to the package doccumentation and help files for a full list of functions.</p>

<h2>
<a id="user-content-6-simulate-nucleotide-sequences-using-phangorn-and-exporting-the-data" class="anchor" href="#6-simulate-nucleotide-sequences-using-phangorn-and-exporting-the-data" aria-hidden="true"><span class="octicon octicon-link"></span></a>6. Simulate nucleotide sequences using phangorn and exporting the data</h2>

<p>We can use the package phangorn to simulate evolvution of nucleotide or amino-acid sequence alignments along a phylogram (the first element of the ratesim object), and save it in an external file in a any format, such as FASTA, for future use.</p>

<ul class="task-list">
<li>Simulate a DNA alignment 2000 base-pairs long, and save it in a file.</li>
</ul>

<div class="highlight highlight-coffee"><pre>sim.dna.data <span class="pl-k">&lt;-</span> simSeq(sim.uncor[[<span class="pl-c1">1</span>]], <span class="pl-v"><span class="pl-v">l <span class="pl-k">=</span></span></span> <span class="pl-c1">2000</span>, <span class="pl-v"><span class="pl-v">type <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>DNA<span class="pl-pds">"</span></span>)
write.phyDat(sim.dna.data, <span class="pl-v"><span class="pl-v">file <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>nelsi_tutorial_dna.fasta<span class="pl-pds">"</span></span>, <span class="pl-v"><span class="pl-v">format <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>fasta<span class="pl-pds">"</span></span>)</pre></div>

<p>Note that the function simSeq can simulate under different models of nucleotide substitution. Use ?simSeq to see details.</p>

<ul class="task-list">
<li>Now save the phylogram in newick format for future reference or comparison, using the ape package.</li>
</ul>

<div class="highlight highlight-coffee"><pre>write.tree(sim.uncor[[<span class="pl-c1">1</span>]], <span class="pl-v"><span class="pl-v">file <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>nelsi_tutorial_pylogram.tree<span class="pl-pds">"</span></span>)</pre></div>

<h2>
<a id="user-content-7-loading-a-virus-data-set-estimated-in-beast-" class="anchor" href="#7-loading-a-virus-data-set-estimated-in-beast-" aria-hidden="true"><span class="octicon octicon-link"></span></a>7. Loading a virus data set estimated in BEAST </h2>

<p>Phylogenetic trees in NEXUS format can have a large number of annotations, with information about rates, times, or other traits for every branch in the tree. These annotations can be read in R with the function read.annotated.nexus, from pacakge <a href="http://www.inside-r.org/packages/cran/epibase">epibase</a>. Then the annotations cat be imported into a tree data matrix to be used with NELSI. Note that epibase is no longer supported, but we have made this function available here.</p>

<p>The examples_data folder contains a tree from <em>ENV</em> sequences from HIV-1 sub-type A collected between 1991 and 2008. </p>

<ul class="task-list">
<li>Type the following code to read the tree (remember to set the working directory to the example_data folder):</li>
</ul>

<div class="highlight highlight-coffee"><pre>hivTree <span class="pl-k">&lt;-</span> read.annotated.nexus(<span class="pl-s"><span class="pl-pds">"</span>hiv_A_env.tree<span class="pl-pds">"</span></span>)</pre></div>

<ul class="task-list">
<li>Plot the tree with the function plot:</li>
</ul>

<div class="highlight highlight-coffee"><pre>plot(hivTree)
axisPhylo()</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-11.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-11.png" alt="plot of chunk unnamed-chunk-11" style="max-width:100%;"></a> </p>

<p>The tree is a chronogram, so that the branch lengths represent units of time. The ages of the nodes can be obtained with the function branching.times, but this only works for ultrametric trees, which is not the case for these data because the samples were obtained at different points in time (heterochronous). The function allnode.times in NELSI can obtain the ages of the nodes and tips for heterochronous trees. The first items of the function are the ages of the tips, and the remaining are the ages of internal nodes.</p>

<ul class="task-list">
<li>Type the following code to plot the tree with out taxon names. Instead add the ages of the tips and internal nodes with the tiplabel and nodelabel functions. </li>
</ul>

<div class="highlight highlight-coffee"><pre>plot(hivTree, <span class="pl-v"><span class="pl-v">show.tip.label <span class="pl-k">=</span></span></span> F)
tip.ages <span class="pl-k">&lt;-</span> round(allnode.times(hivTree), <span class="pl-c1">2</span>)  <span class="pl-c"># Round to two decimal places for a clearer plot</span>
<span class="pl-c"># See the tip ages. The first 19 elements are the ages of the tips (the tree</span>
<span class="pl-c"># has 19 tips), while the remaining are the ages of internal nodes</span>
tiplabels(tip.ages[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">19</span>])
nodelabels(tip.ages[<span class="pl-c1">20</span><span class="pl-k">:</span><span class="pl-c1">37</span>])
axisPhylo()</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-12.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-12.png" alt="plot of chunk unnamed-chunk-12" style="max-width:100%;"></a> </p>

<p>The age of the youngest tip is always assigned an age of 0. The age of the root is calculated with reference to the youngest tip.</p>

<h2>
<a id="user-content-8-obtaining-the-tree-data-matrix-for-a-tree-estimated-in-beast" class="anchor" href="#8-obtaining-the-tree-data-matrix-for-a-tree-estimated-in-beast" aria-hidden="true"><span class="octicon octicon-link"></span></a>8. Obtaining the tree data matrix for a tree estimated in BEAST</h2>

<p>We will use the tree in 7. to obtain the tree data matrix and plot the rates through time.</p>

<ul class="task-list">
<li>Use the function get.tree.data.matrix for the HIV tree in 7 and inspect the tree data matrix:</li>
</ul>

<div class="highlight highlight-coffee"><pre>hivDataMatrix <span class="pl-k">&lt;-</span> as.data.frame(trann2trdat(hivTree))
head(hivDataMatrix)</pre></div>

<div class="highlight highlight-coffee"><pre><span class="pl-c">##   branch parent daughter midage     rate blensubs blentime</span>
<span class="pl-c">## 1      1     20       21 154.55 0.001301  0.02622   20.159</span>
<span class="pl-c">## 2      2     21       22  99.35 0.001314  0.12836   97.721</span>
<span class="pl-c">## 3      3     22        1  32.00 0.001286  0.05929   46.113</span>
<span class="pl-c">## 4      4     22       23  49.50 0.001274  0.01112    8.730</span>
<span class="pl-c">## 5      5     23        2  31.84 0.001332  0.03859   28.980</span>
<span class="pl-c">## 6      6     23       24  41.23 0.001291  0.01154    8.936</span></pre></div>

<h2>
<a id="user-content-9-root-to-tip-regressions-for-trees-estimated-in-beast" class="anchor" href="#9-root-to-tip-regressions-for-trees-estimated-in-beast" aria-hidden="true"><span class="octicon octicon-link"></span></a>9. Root-to-tip regressions for trees estimated in BEAST</h2>

<p>For heterochronous data one can test the molecular clock by conducting a regression of the number of substitutions from the root to the tips vs. the time from the root to the tip, like in the program <a href="http://tree.bio.ed.ac.uk/software/pathogen/">Path-o-Gen</a> (Rambaut <em>et al.</em> 2009). With the help of a few functions from NELSI, we can conduct these analyses in R.</p>

<ul class="task-list">
<li>Obtain the ages of the tips with the function allnode.times with the HIV chronogram. Specify the argument tipsonly = T, which will return the ages of the tips, and not those of internal nodes.</li>
</ul>

<div class="highlight highlight-coffee"><pre>tipsTimes <span class="pl-k">&lt;-</span> allnode.times(hivTree, <span class="pl-v"><span class="pl-v">tipsonly <span class="pl-k">=</span></span></span> T)</pre></div>

<ul class="task-list">
<li>We can use the tree data matrix from 8. to obtan the HIV phylogram (with branch lengths in substitutions). To do this, create a copy of the chronogram in variable hivPhylogram and set the branch lengths to the number of substitutions from the tree data matrix:</li>
</ul>

<div class="highlight highlight-coffee"><pre>hivPhylogram <span class="pl-k">&lt;-</span> hivTree
hivPhylogram$edge.length <span class="pl-k">&lt;-</span> hivDataMatrix$blensubs</pre></div>

<ul class="task-list">
<li>The root-to-tip distances in the phylogram are the number of substutions from the root to the tips. Save this in an other variable:</li>
</ul>

<div class="highlight highlight-coffee"><pre>tipsSubstitutions <span class="pl-k">&lt;-</span> allnode.times(hivPhylogram, <span class="pl-v"><span class="pl-v">tipsonly <span class="pl-k">=</span></span></span> T)</pre></div>

<ul class="task-list">
<li>The variables tipsTimes and tipsSubstitutions can be used to plot the data and test the linear regression with basic linear models:</li>
</ul>

<div class="highlight highlight-coffee"><pre>plot(tipsTimes, tipsSubstitutions, <span class="pl-v"><span class="pl-v">pch <span class="pl-k">=</span></span></span> <span class="pl-c1">20</span>, <span class="pl-v"><span class="pl-v">ylab <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>Substitutions from the root to tips (substitutions)<span class="pl-pds">"</span></span>, 
    <span class="pl-v"><span class="pl-v">xlab <span class="pl-k">=</span></span></span> <span class="pl-s"><span class="pl-pds">"</span>Time from the root to the tip (years)<span class="pl-pds">"</span></span>)
hivRegression <span class="pl-k">&lt;-</span> lm(tipsSubstitutions <span class="pl-k">~</span> tipsTimes)
abline(hivRegression)</pre></div>

<p><a href="/sebastianduchene/NELSI/blob/master/figure/unnamed-chunk-17.png" target="_blank"><img src="/sebastianduchene/NELSI/raw/master/figure/unnamed-chunk-17.png" alt="plot of chunk unnamed-chunk-17" style="max-width:100%;"></a> </p>

<div class="highlight highlight-coffee"><pre>summary(hivRegression)</pre></div>

<div class="highlight highlight-coffee"><pre><span class="pl-c">## </span>
<span class="pl-c">## Call:</span>
<span class="pl-c">## lm(formula = tipsSubstitutions ~ tipsTimes)</span>
<span class="pl-c">## </span>
<span class="pl-c">## Residuals:</span>
<span class="pl-c">##       Min        1Q    Median        3Q       Max </span>
<span class="pl-c">## -0.007700 -0.000927  0.000179  0.002335  0.006609 </span>
<span class="pl-c">## </span>
<span class="pl-c">## Coefficients:</span>
<span class="pl-c">##             Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="pl-c">## (Intercept) 0.000912   0.002026    0.45     0.66    </span>
<span class="pl-c">## tipsTimes   0.001233   0.000189    6.52  5.3e-06 ***</span>
<span class="pl-c">## ---</span>
<span class="pl-c">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="pl-c">## </span>
<span class="pl-c">## Residual standard error: 0.00386 on 17 degrees of freedom</span>
<span class="pl-c">## Multiple R-squared:  0.714,  Adjusted R-squared:  0.697 </span>
<span class="pl-c">## F-statistic: 42.5 on 1 and 17 DF,  p-value: 5.25e-06</span></pre></div>

<p>In this case it the data appear to have clock-like behaviour.</p>

<h2>
<a id="user-content-references" class="anchor" href="#references" aria-hidden="true"><span class="octicon octicon-link"></span></a>References</h2>

<p>Drummond, A. J., Ho, S. Y., Phillips, M. J., &amp; Rambaut, A. (2006). Relaxed phylogenetics and dating with confidence. <em>PLOS Biology</em>, 4(5), e88.</p>

<p>Kishino, H., Thorne, J. L., &amp; Bruno, W. J. (2001). Performance of a divergence time estimation method under a probabilistic model of rate evolution. <em>Molecular Biology and Evolution</em>, 18(3), 352-361.</p>

<p>Hasegawa, M., Kishino, H., &amp; Yano, T. A. (1989). Estimation of branching dates among primates by molecular clocks of nuclear DNA which slowed down in Hominoidea. <em>Journal of Human Evolution</em>, 18(5), 461-476.</p>

<p>Heath, T. A., Holder, M. T., &amp; Huelsenbeck, J. P. (2012). A dirichlet process prior for estimating lineage-specific substitution rates. <em>Molecular Biology and Evolution</em>, 29(3), 939-955.</p>

<p>Ho, S. Y., Phillips, M. J., Drummond, A. J., &amp; Cooper, A. (2005). Accuracy of rate estimation using relaxed-clock models with a critical focus on the early metazoan radiation. <em>Molecular Biology and Evolution</em>, 22(5), 1355-1363.</p>

<p>Lepage, T., Bryant, D., Philippe, H., &amp; Lartillot, N. (2007). A general comparison of relaxed molecular clock models. <em>Molecular Biology and Evolution</em>, 24(12), 2669-2680.</p>

<p>Popescu, A. A., Huber, K. T., &amp; Paradis, E. (2012). Ape 3.0: New tools for distance-based phylogenetics and evolutionary analysis in R. <em>Bioinformatics</em>, 28(11), 1536-1537.</p>

<p>Rambaut, A. (2009). Path-O-Gen: temporal signal investigation tool.</p>

<p>Schliep, K. P. (2011). phangorn: Phylogenetic analysis in R. <em>Bioinformatics</em>, 27(4), 592-593.</p>

<p>Thorne, J.L., Kishino, H., and Painter, I.S., Estimating the rate of evolution of the rate of molecular evolution. <em>Molecular Biology and Evolution</em> 15.12 (1998): 1647-1657.</p>

<p>Yoder, A. D., &amp; Yang, Z. (2000). Estimation of primate speciation dates using local molecular clocks. <em>Molecular Biology and Evolution</em>, 17(7), 1081-1090.</p>

<p>Zuckerkandl,E. and Pauling,L. (1962) Molecular disease, evolution, and genetic heterogeneity. In: Kasha,M. and Pullman,B. (eds) Horizons in Biochemistry. Academic Press, New York, pp. 189–225.</p>

<h2>
<a id="user-content-bugs-and-version-history" class="anchor" href="#bugs-and-version-history" aria-hidden="true"><span class="octicon octicon-link"></span></a>Bugs and version history</h2>

<p>6 November 2014</p>

<ul class="task-list">
<li>0.21 - Removed phangorn dependency in allnode.times</li>
</ul>
</article>
  </div>

</div>

<a href="#jump-to-line" rel="facebox[.linejump]" data-hotkey="l" style="display:none">Jump to Line</a>
<div id="jump-to-line" style="display:none">
  <form accept-charset="UTF-8" class="js-jump-to-line-form">
    <input class="linejump-input js-jump-to-line-field" type="text" placeholder="Jump to line&hellip;" autofocus>
    <button type="submit" class="btn">Go</button>
  </form>
</div>

        </div>

      </div><!-- /.repo-container -->
      <div class="modal-backdrop"></div>
    </div><!-- /.container -->
  </div><!-- /.site -->


    </div><!-- /.wrapper -->

      <div class="container">
  <div class="site-footer" role="contentinfo">
    <ul class="site-footer-links right">
        <li><a href="https://status.github.com/" data-ga-click="Footer, go to status, text:status">Status</a></li>
      <li><a href="https://developer.github.com" data-ga-click="Footer, go to api, text:api">API</a></li>
      <li><a href="https://training.github.com" data-ga-click="Footer, go to training, text:training">Training</a></li>
      <li><a href="https://shop.github.com" data-ga-click="Footer, go to shop, text:shop">Shop</a></li>
        <li><a href="https://github.com/blog" data-ga-click="Footer, go to blog, text:blog">Blog</a></li>
        <li><a href="https://github.com/about" data-ga-click="Footer, go to about, text:about">About</a></li>

    </ul>

    <a href="https://github.com" aria-label="Homepage">
      <span class="mega-octicon octicon-mark-github" title="GitHub"></span>
</a>
    <ul class="site-footer-links">
      <li>&copy; 2015 <span title="0.03309s from github-fe118-cp1-prd.iad.github.net">GitHub</span>, Inc.</li>
        <li><a href="https://github.com/site/terms" data-ga-click="Footer, go to terms, text:terms">Terms</a></li>
        <li><a href="https://github.com/site/privacy" data-ga-click="Footer, go to privacy, text:privacy">Privacy</a></li>
        <li><a href="https://github.com/security" data-ga-click="Footer, go to security, text:security">Security</a></li>
        <li><a href="https://github.com/contact" data-ga-click="Footer, go to contact, text:contact">Contact</a></li>
    </ul>
  </div>
</div>


    <div class="fullscreen-overlay js-fullscreen-overlay" id="fullscreen_overlay">
  <div class="fullscreen-container js-suggester-container">
    <div class="textarea-wrap">
      <textarea name="fullscreen-contents" id="fullscreen-contents" class="fullscreen-contents js-fullscreen-contents" placeholder=""></textarea>
      <div class="suggester-container">
        <div class="suggester fullscreen-suggester js-suggester js-navigation-container"></div>
      </div>
    </div>
  </div>
  <div class="fullscreen-sidebar">
    <a href="#" class="exit-fullscreen js-exit-fullscreen tooltipped tooltipped-w" aria-label="Exit Zen Mode">
      <span class="mega-octicon octicon-screen-normal"></span>
    </a>
    <a href="#" class="theme-switcher js-theme-switcher tooltipped tooltipped-w"
      aria-label="Switch themes">
      <span class="octicon octicon-color-mode"></span>
    </a>
  </div>
</div>



    
    

    <div id="ajax-error-message" class="flash flash-error">
      <span class="octicon octicon-alert"></span>
      <a href="#" class="octicon octicon-x flash-close js-ajax-error-dismiss" aria-label="Dismiss error"></a>
      Something went wrong with that request. Please try again.
    </div>


      <script crossorigin="anonymous" src="https://assets-cdn.github.com/assets/frameworks-d22b59d0085e83b7549ba4341ec9e68f80c2f29c8e49213ee182003dc8d568c6.js"></script>
      <script async="async" crossorigin="anonymous" src="https://assets-cdn.github.com/assets/github-0bc0f45c838b5d9d25bc071d2a4b0abe759a093392087dce55cd2caa00ea4f36.js"></script>
      
      

  </body>
</html>

